<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Helper Functions for LIBD Deconvolution â€¢ DeconvoBuddies</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><link href="extra.css" rel="stylesheet">
<meta property="og:title" content="Helper Functions for LIBD Deconvolution">
<meta property="og:description" content="Funtions helpful for LIBD deconvolution project. Includes tools for marker finding with mean ratio, expression plotting, and plotting deconvolution results. Working to include DLPFC datasets.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">DeconvoBuddies</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="articles/DeconvoBuddies.html">Get started</a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/Deconvolution_Benchmark_DLPFC.html">Deconvolution_Benchmark_DLPFC</a>
    </li>
    <li>
      <a href="articles/Deconvolution_with_Bisque.html">Deconvolution_with_Bisque</a>
    </li>
  </ul>
</li>
<li>
  <a href="news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/lahuuki/DeconvoBuddies/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
<div class="section level1">
<div class="page-header"><h1 id="deconvobuddies">DeconvoBuddies<a class="anchor" aria-label="anchor" href="#deconvobuddies"></a>
</h1></div>
<!-- badges: start -->

<p>The goal of <code>DeconvoBuddies</code> is to provide helper functions for the deconvolution process</p>
<div class="section level2">
<h2 id="installation-instructions">Installation instructions<a class="anchor" aria-label="anchor" href="#installation-instructions"></a>
</h2>
<p>Get the latest stable <code>R</code> release from <a href="http://cran.r-project.org/" class="external-link">CRAN</a>. Then install <code>DeconvoBuddies</code> using from <a href="http://bioconductor.org/" class="external-link">Bioconductor</a> the following code:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"DeconvoBuddies"</span><span class="op">)</span></span></code></pre></div>
<p>And the development version from <a href="https://github.com/" class="external-link">GitHub</a> with:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"LieberInstitute/DeconvoBuddies"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="example">Example<a class="anchor" aria-label="anchor" href="#example"></a>
</h2>
<div class="sourceCode" id="cb3"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a><span class="co">#&gt; Attaching package: 'dplyr'</span></span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a><span class="co">#&gt;     filter, lag</span></span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb3-8"><a href="#cb3-8" tabindex="-1"></a><span class="co">#&gt;     intersect, setdiff, setequal, union</span></span>
<span id="cb3-9"><a href="#cb3-9" tabindex="-1"></a><span class="co">#&gt; Loading required package: SummarizedExperiment</span></span>
<span id="cb3-10"><a href="#cb3-10" tabindex="-1"></a><span class="co">#&gt; Loading required package: MatrixGenerics</span></span>
<span id="cb3-11"><a href="#cb3-11" tabindex="-1"></a><span class="co">#&gt; Loading required package: matrixStats</span></span>
<span id="cb3-12"><a href="#cb3-12" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb3-13"><a href="#cb3-13" tabindex="-1"></a><span class="co">#&gt; Attaching package: 'matrixStats'</span></span>
<span id="cb3-14"><a href="#cb3-14" tabindex="-1"></a><span class="co">#&gt; The following object is masked from 'package:dplyr':</span></span>
<span id="cb3-15"><a href="#cb3-15" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb3-16"><a href="#cb3-16" tabindex="-1"></a><span class="co">#&gt;     count</span></span>
<span id="cb3-17"><a href="#cb3-17" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb3-18"><a href="#cb3-18" tabindex="-1"></a><span class="co">#&gt; Attaching package: 'MatrixGenerics'</span></span>
<span id="cb3-19"><a href="#cb3-19" tabindex="-1"></a><span class="co">#&gt; The following objects are masked from 'package:matrixStats':</span></span>
<span id="cb3-20"><a href="#cb3-20" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb3-21"><a href="#cb3-21" tabindex="-1"></a><span class="co">#&gt;     colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,</span></span>
<span id="cb3-22"><a href="#cb3-22" tabindex="-1"></a><span class="co">#&gt;     colCounts, colCummaxs, colCummins, colCumprods, colCumsums,</span></span>
<span id="cb3-23"><a href="#cb3-23" tabindex="-1"></a><span class="co">#&gt;     colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,</span></span>
<span id="cb3-24"><a href="#cb3-24" tabindex="-1"></a><span class="co">#&gt;     colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,</span></span>
<span id="cb3-25"><a href="#cb3-25" tabindex="-1"></a><span class="co">#&gt;     colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,</span></span>
<span id="cb3-26"><a href="#cb3-26" tabindex="-1"></a><span class="co">#&gt;     colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,</span></span>
<span id="cb3-27"><a href="#cb3-27" tabindex="-1"></a><span class="co">#&gt;     colWeightedMeans, colWeightedMedians, colWeightedSds,</span></span>
<span id="cb3-28"><a href="#cb3-28" tabindex="-1"></a><span class="co">#&gt;     colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,</span></span>
<span id="cb3-29"><a href="#cb3-29" tabindex="-1"></a><span class="co">#&gt;     rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,</span></span>
<span id="cb3-30"><a href="#cb3-30" tabindex="-1"></a><span class="co">#&gt;     rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,</span></span>
<span id="cb3-31"><a href="#cb3-31" tabindex="-1"></a><span class="co">#&gt;     rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,</span></span>
<span id="cb3-32"><a href="#cb3-32" tabindex="-1"></a><span class="co">#&gt;     rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,</span></span>
<span id="cb3-33"><a href="#cb3-33" tabindex="-1"></a><span class="co">#&gt;     rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,</span></span>
<span id="cb3-34"><a href="#cb3-34" tabindex="-1"></a><span class="co">#&gt;     rowWeightedMads, rowWeightedMeans, rowWeightedMedians,</span></span>
<span id="cb3-35"><a href="#cb3-35" tabindex="-1"></a><span class="co">#&gt;     rowWeightedSds, rowWeightedVars</span></span>
<span id="cb3-36"><a href="#cb3-36" tabindex="-1"></a><span class="co">#&gt; Loading required package: GenomicRanges</span></span>
<span id="cb3-37"><a href="#cb3-37" tabindex="-1"></a><span class="co">#&gt; Loading required package: stats4</span></span>
<span id="cb3-38"><a href="#cb3-38" tabindex="-1"></a><span class="co">#&gt; Loading required package: BiocGenerics</span></span>
<span id="cb3-39"><a href="#cb3-39" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb3-40"><a href="#cb3-40" tabindex="-1"></a><span class="co">#&gt; Attaching package: 'BiocGenerics'</span></span>
<span id="cb3-41"><a href="#cb3-41" tabindex="-1"></a><span class="co">#&gt; The following objects are masked from 'package:dplyr':</span></span>
<span id="cb3-42"><a href="#cb3-42" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb3-43"><a href="#cb3-43" tabindex="-1"></a><span class="co">#&gt;     combine, intersect, setdiff, union</span></span>
<span id="cb3-44"><a href="#cb3-44" tabindex="-1"></a><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span id="cb3-45"><a href="#cb3-45" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb3-46"><a href="#cb3-46" tabindex="-1"></a><span class="co">#&gt;     IQR, mad, sd, var, xtabs</span></span>
<span id="cb3-47"><a href="#cb3-47" tabindex="-1"></a><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span id="cb3-48"><a href="#cb3-48" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb3-49"><a href="#cb3-49" tabindex="-1"></a><span class="co">#&gt;     anyDuplicated, aperm, append, as.data.frame, basename, cbind,</span></span>
<span id="cb3-50"><a href="#cb3-50" tabindex="-1"></a><span class="co">#&gt;     colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,</span></span>
<span id="cb3-51"><a href="#cb3-51" tabindex="-1"></a><span class="co">#&gt;     get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,</span></span>
<span id="cb3-52"><a href="#cb3-52" tabindex="-1"></a><span class="co">#&gt;     match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,</span></span>
<span id="cb3-53"><a href="#cb3-53" tabindex="-1"></a><span class="co">#&gt;     Position, rank, rbind, Reduce, rownames, sapply, setdiff, sort,</span></span>
<span id="cb3-54"><a href="#cb3-54" tabindex="-1"></a><span class="co">#&gt;     table, tapply, union, unique, unsplit, which.max, which.min</span></span>
<span id="cb3-55"><a href="#cb3-55" tabindex="-1"></a><span class="co">#&gt; Loading required package: S4Vectors</span></span>
<span id="cb3-56"><a href="#cb3-56" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb3-57"><a href="#cb3-57" tabindex="-1"></a><span class="co">#&gt; Attaching package: 'S4Vectors'</span></span>
<span id="cb3-58"><a href="#cb3-58" tabindex="-1"></a><span class="co">#&gt; The following objects are masked from 'package:dplyr':</span></span>
<span id="cb3-59"><a href="#cb3-59" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb3-60"><a href="#cb3-60" tabindex="-1"></a><span class="co">#&gt;     first, rename</span></span>
<span id="cb3-61"><a href="#cb3-61" tabindex="-1"></a><span class="co">#&gt; The following object is masked from 'package:utils':</span></span>
<span id="cb3-62"><a href="#cb3-62" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb3-63"><a href="#cb3-63" tabindex="-1"></a><span class="co">#&gt;     findMatches</span></span>
<span id="cb3-64"><a href="#cb3-64" tabindex="-1"></a><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span id="cb3-65"><a href="#cb3-65" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb3-66"><a href="#cb3-66" tabindex="-1"></a><span class="co">#&gt;     expand.grid, I, unname</span></span>
<span id="cb3-67"><a href="#cb3-67" tabindex="-1"></a><span class="co">#&gt; Loading required package: IRanges</span></span>
<span id="cb3-68"><a href="#cb3-68" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb3-69"><a href="#cb3-69" tabindex="-1"></a><span class="co">#&gt; Attaching package: 'IRanges'</span></span>
<span id="cb3-70"><a href="#cb3-70" tabindex="-1"></a><span class="co">#&gt; The following objects are masked from 'package:dplyr':</span></span>
<span id="cb3-71"><a href="#cb3-71" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb3-72"><a href="#cb3-72" tabindex="-1"></a><span class="co">#&gt;     collapse, desc, slice</span></span>
<span id="cb3-73"><a href="#cb3-73" tabindex="-1"></a><span class="co">#&gt; Loading required package: GenomeInfoDb</span></span>
<span id="cb3-74"><a href="#cb3-74" tabindex="-1"></a><span class="co">#&gt; Loading required package: Biobase</span></span>
<span id="cb3-75"><a href="#cb3-75" tabindex="-1"></a><span class="co">#&gt; Welcome to Bioconductor</span></span>
<span id="cb3-76"><a href="#cb3-76" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb3-77"><a href="#cb3-77" tabindex="-1"></a><span class="co">#&gt;     Vignettes contain introductory material; view with</span></span>
<span id="cb3-78"><a href="#cb3-78" tabindex="-1"></a><span class="co">#&gt;     'browseVignettes()'. To cite Bioconductor, see</span></span>
<span id="cb3-79"><a href="#cb3-79" tabindex="-1"></a><span class="co">#&gt;     'citation("Biobase")', and for packages 'citation("pkgname")'.</span></span>
<span id="cb3-80"><a href="#cb3-80" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb3-81"><a href="#cb3-81" tabindex="-1"></a><span class="co">#&gt; Attaching package: 'Biobase'</span></span>
<span id="cb3-82"><a href="#cb3-82" tabindex="-1"></a><span class="co">#&gt; The following object is masked from 'package:MatrixGenerics':</span></span>
<span id="cb3-83"><a href="#cb3-83" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb3-84"><a href="#cb3-84" tabindex="-1"></a><span class="co">#&gt;     rowMedians</span></span>
<span id="cb3-85"><a href="#cb3-85" tabindex="-1"></a><span class="co">#&gt; The following objects are masked from 'package:matrixStats':</span></span>
<span id="cb3-86"><a href="#cb3-86" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb3-87"><a href="#cb3-87" tabindex="-1"></a><span class="co">#&gt;     anyMissing, rowMedians</span></span></code></pre></div>
<p>Get mean ratios for each gene x cell type</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ratios</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/get_mean_ratio2.html">get_mean_ratio2</a></span><span class="op">(</span><span class="va">sce.test</span><span class="op">)</span></span>
<span><span class="va">fc</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/findMarkers_1vAll.html">findMarkers_1vAll</a></span><span class="op">(</span><span class="va">sce.test</span><span class="op">)</span></span>
<span><span class="co">#&gt; Inhib.2 - '2024-03-19 16:13:23.361677</span></span>
<span><span class="co">#&gt; Inhib.1 - '2024-03-19 16:13:23.654662</span></span>
<span><span class="co">#&gt; OPC - '2024-03-19 16:13:23.82545</span></span>
<span><span class="co">#&gt; Astro - '2024-03-19 16:13:24.033885</span></span>
<span><span class="co">#&gt; Excit.2 - '2024-03-19 16:13:24.268223</span></span>
<span><span class="co">#&gt; Oligo - '2024-03-19 16:13:24.432648</span></span>
<span><span class="co">#&gt; Micro - '2024-03-19 16:13:24.600515</span></span>
<span><span class="co">#&gt; Excit.1 - '2024-03-19 16:13:24.768858</span></span>
<span><span class="co">#&gt; Building Table - 2024-03-19 16:13:24.950633</span></span>
<span><span class="co">#&gt; ** Done! **</span></span>
<span></span>
<span><span class="op">(</span><span class="va">marker_stats</span> <span class="op">&lt;-</span> <span class="fu">left_join</span><span class="op">(</span><span class="va">ratios</span>, <span class="va">fc</span>, by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"gene"</span>, <span class="st">"cellType.target"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; # A tibble: 1,778 Ã— 15</span></span>
<span><span class="co">#&gt;    gene       cellType.target mean.target cellType  mean ratio rank_ratio Symbol</span></span>
<span><span class="co">#&gt;    &lt;chr&gt;      &lt;fct&gt;                 &lt;dbl&gt; &lt;fct&gt;    &lt;dbl&gt; &lt;dbl&gt;      &lt;int&gt; &lt;chr&gt; </span></span>
<span><span class="co">#&gt;  1 ENSG00000â€¦ Inhib.2               1.00  Excit.2  0.239  4.20          1 AL139â€¦</span></span>
<span><span class="co">#&gt;  2 ENSG00000â€¦ Inhib.2               1.71  Astro    0.512  3.35          2 SDC3  </span></span>
<span><span class="co">#&gt;  3 ENSG00000â€¦ Inhib.2               0.950 Astro    0.413  2.30          3 IFI44 </span></span>
<span><span class="co">#&gt;  4 ENSG00000â€¦ Inhib.2               3.32  Astro    1.47   2.26          4 COL11â€¦</span></span>
<span><span class="co">#&gt;  5 ENSG00000â€¦ Inhib.2               3.55  Astro    1.62   2.19          5 NTNG1 </span></span>
<span><span class="co">#&gt;  6 ENSG00000â€¦ Inhib.2               1.22  Excit.1  0.560  2.18          6 TRIM62</span></span>
<span><span class="co">#&gt;  7 ENSG00000â€¦ Inhib.2               3.37  Excit.2  1.96   1.72          7 USP24 </span></span>
<span><span class="co">#&gt;  8 ENSG00000â€¦ Inhib.2               3.42  Inhib.1  2.44   1.40          8 SPATA6</span></span>
<span><span class="co">#&gt;  9 ENSG00000â€¦ Inhib.2               1.21  Astro    0.914  1.33          9 ABCD3 </span></span>
<span><span class="co">#&gt; 10 ENSG00000â€¦ Inhib.2               1.21  Astro    0.920  1.32         10 GNG12 </span></span>
<span><span class="co">#&gt; # â„¹ 1,768 more rows</span></span>
<span><span class="co">#&gt; # â„¹ 7 more variables: anno_ratio &lt;chr&gt;, logFC &lt;dbl&gt;, log.p.value &lt;dbl&gt;,</span></span>
<span><span class="co">#&gt; #   log.FDR &lt;dbl&gt;, std.logFC &lt;dbl&gt;, rank_marker &lt;int&gt;, anno_logFC &lt;chr&gt;</span></span></code></pre></div>
<div class="section level3">
<h3 id="extablish-color-scheme">Extablish Color Scheme<a class="anchor" aria-label="anchor" href="#extablish-color-scheme"></a>
</h3>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cell_types</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/levels.html" class="external-link">levels</a></span><span class="op">(</span><span class="va">sce.test</span><span class="op">$</span><span class="va">cellType</span><span class="op">)</span></span>
<span><span class="va">cell_colors</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/create_cell_colors.html">create_cell_colors</a></span><span class="op">(</span>cell_types <span class="op">=</span> <span class="va">cell_types</span>, pallet <span class="op">=</span> <span class="st">"classic"</span>, split <span class="op">=</span> <span class="st">"\\."</span>, preview <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-establish_colors-1.png" width="100%"></p>
</div>
<div class="section level3">
<h3 id="plot-expression-of-specified-genes">Plot Expression of Specified Genes<a class="anchor" aria-label="anchor" href="#plot-expression-of-specified-genes"></a>
</h3>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sce_symbol</span> <span class="op">&lt;-</span> <span class="va">sce.test</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">sce_symbol</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">rowData</span><span class="op">(</span><span class="va">sce.test</span><span class="op">)</span><span class="op">$</span><span class="va">Symbol</span></span>
<span><span class="fu"><a href="reference/plot_gene_express.html">plot_gene_express</a></span><span class="op">(</span>sce <span class="op">=</span> <span class="va">sce_symbol</span>, genes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"RNF220"</span>, <span class="st">"CSF3R"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-plot_gene_expression-1.png" width="100%"></p>
</div>
<div class="section level3">
<h3 id="plot-expression-of-marker-genes">Plot Expression of Marker Genes<a class="anchor" aria-label="anchor" href="#plot-expression-of-marker-genes"></a>
</h3>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/plot_marker_express.html">plot_marker_express</a></span><span class="op">(</span><span class="va">sce.test</span>,</span>
<span>    <span class="va">marker_stats</span>,</span>
<span>    <span class="st">"Astro"</span>,</span>
<span>    n_genes <span class="op">=</span> <span class="fl">5</span>,</span>
<span>    rank_col <span class="op">=</span> <span class="st">"rank_ratio"</span>,</span>
<span>    anno_col <span class="op">=</span> <span class="st">"anno_ratio"</span>,</span>
<span>    color_pal <span class="op">=</span> <span class="va">cell_colors</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-plot_marker_expression-1.png" width="100%"></p>
</div>
<div class="section level3">
<h3 id="create-composition-bar-plot">Create Composition Bar Plot<a class="anchor" aria-label="anchor" href="#create-composition-bar-plot"></a>
</h3>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pd</span> <span class="op">&lt;-</span> <span class="fu">SummarizedExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_bulk_test</span><span class="op">)</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">est_prop_long</span> <span class="op">&lt;-</span> <span class="va">est_prop</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">rownames_to_column</a></span><span class="op">(</span><span class="st">"RNum"</span><span class="op">)</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu">tidyr</span><span class="fu">::</span><span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_longer.html" class="external-link">pivot_longer</a></span><span class="op">(</span><span class="op">!</span><span class="va">RNum</span>, names_to <span class="op">=</span> <span class="st">"cell_type"</span>, values_to <span class="op">=</span> <span class="st">"prop"</span><span class="op">)</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span><span class="va">pd</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">RNum</span>, <span class="va">Dx</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>a <span class="op">=</span> <span class="st">"a"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Joining with `by = join_by(RNum)`</span></span>
<span></span>
<span><span class="fu"><a href="reference/plot_composition_bar.html">plot_composition_bar</a></span><span class="op">(</span><span class="va">est_prop_long</span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-composition_bar_plot-1.png" width="100%"></p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/plot_composition_bar.html">plot_composition_bar</a></span><span class="op">(</span><span class="va">est_prop_long</span>, x_col <span class="op">=</span> <span class="st">"Dx"</span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-composition_bar_plot-2.png" width="100%"></p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/plot_composition_bar.html">plot_composition_bar</a></span><span class="op">(</span><span class="va">est_prop_long</span>, x_col <span class="op">=</span> <span class="st">"Dx"</span>, min_prop_text <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-composition_bar_plot-3.png" width="100%"></p>
</div>
</div>
<div class="section level2">
<h2 id="citation">Citation<a class="anchor" aria-label="anchor" href="#citation"></a>
</h2>
<p>Below is the citation output from using <code>citation('DeconvoBuddies')</code> in R. Please run this yourself to check for any updates on how to cite <strong>DeconvoBuddies</strong>.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/citation.html" class="external-link">citation</a></span><span class="op">(</span><span class="st">"DeconvoBuddies"</span><span class="op">)</span>, bibtex <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; To cite package 'DeconvoBuddies' in publications use:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Huuki-Myers LA, Maynard KR, Hicks SC, Zandi P, Kleinman JE, Hyde TM,</span></span>
<span><span class="co">#&gt;   Goes FS, Collado-Torres L (2024). _DeconvoBuddies: a R/Bioconductor</span></span>
<span><span class="co">#&gt;   package with deconvolution helper functions_.</span></span>
<span><span class="co">#&gt;   doi:10.18129/B9.bioc.DeconvoBuddies</span></span>
<span><span class="co">#&gt;   &lt;https://doi.org/10.18129/B9.bioc.DeconvoBuddies&gt;,</span></span>
<span><span class="co">#&gt;   https://github.com/LieberInstitute/DeconvoBuddies/DeconvoBuddies - R</span></span>
<span><span class="co">#&gt;   package version 0.99.0,</span></span>
<span><span class="co">#&gt;   &lt;http://www.bioconductor.org/packages/DeconvoBuddies&gt;.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; A BibTeX entry for LaTeX users is</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   @Manual{,</span></span>
<span><span class="co">#&gt;     title = {DeconvoBuddies: a R/Bioconductor package with deconvolution helper functions},</span></span>
<span><span class="co">#&gt;     author = {Louise A. Huuki-Myers and Kristen R. Maynard and Stephanie C. Hicks and Peter Zandi and Jole E. Kleinman and Tom M. Hyde and Fernando S. Goes and Leonardo Collado-Torres},</span></span>
<span><span class="co">#&gt;     year = {2024},</span></span>
<span><span class="co">#&gt;     url = {http://www.bioconductor.org/packages/DeconvoBuddies},</span></span>
<span><span class="co">#&gt;     note = {https://github.com/LieberInstitute/DeconvoBuddies/DeconvoBuddies - R package version 0.99.0},</span></span>
<span><span class="co">#&gt;     doi = {10.18129/B9.bioc.DeconvoBuddies},</span></span>
<span><span class="co">#&gt;   }</span></span></code></pre></div>
<p>Please note that the <code>DeconvoBuddies</code> was only made possible thanks to many other R and bioinformatics software authors, which are cited either in the vignettes and/or the paper(s) describing this package.</p>
</div>
<div class="section level2">
<h2 id="code-of-conduct">Code of Conduct<a class="anchor" aria-label="anchor" href="#code-of-conduct"></a>
</h2>
<p>Please note that the <code>DeconvoBuddies</code> project is released with a <a href="https://contributor-covenant.org/version/2/0/CODE_OF_CONDUCT.html" class="external-link">Contributor Code of Conduct</a>. By contributing to this project, you agree to abide by its terms.</p>
</div>
<div class="section level2">
<h2 id="development-tools">Development tools<a class="anchor" aria-label="anchor" href="#development-tools"></a>
</h2>
<ul>
<li>Continuous code testing is possible thanks to <a href="https://www.tidyverse.org/blog/2020/04/usethis-1-6-0/" class="external-link">GitHub actions</a> through <em><a href="https://CRAN.R-project.org/package=usethis" class="external-link">usethis</a></em>, <em><a href="https://CRAN.R-project.org/package=remotes" class="external-link">remotes</a></em>, and <em><a href="https://CRAN.R-project.org/package=rcmdcheck" class="external-link">rcmdcheck</a></em> customized to use <a href="https://www.bioconductor.org/help/docker/" class="external-link">Bioconductorâ€™s docker containers</a> and <em><a href="https://bioconductor.org/packages/3.17/BiocCheck" class="external-link">BiocCheck</a></em>.</li>
<li>Code coverage assessment is possible thanks to <a href="https://codecov.io/gh" class="external-link">codecov</a> and <em><a href="https://CRAN.R-project.org/package=covr" class="external-link">covr</a></em>.</li>
<li>The <a href="http://.github.io/DeconvoBuddies" class="external-link">documentation website</a> is automatically updated thanks to <em><a href="https://CRAN.R-project.org/package=pkgdown" class="external-link">pkgdown</a></em>.</li>
<li>The code is styled automatically thanks to <em><a href="https://CRAN.R-project.org/package=styler" class="external-link">styler</a></em>.</li>
<li>The documentation is formatted thanks to <em><a href="https://CRAN.R-project.org/package=devtools" class="external-link">devtools</a></em> and <em><a href="https://CRAN.R-project.org/package=roxygen2" class="external-link">roxygen2</a></em>.</li>
</ul>
<p>For more details, check the <code>dev</code> directory.</p>
<p>This package was developed using <em><a href="https://bioconductor.org/packages/3.17/biocthis" class="external-link">biocthis</a></em>.</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/lahuuki/DeconvoBuddies/" class="external-link">Browse source code</a></li>
<li><a href="https://support.bioconductor.org/t/DeconvoBuddies" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/Artistic-2.0" class="external-link">Artistic-2.0</a></li>
</ul>
</div>

<div class="community">
<h2 data-toc-skip>Community</h2>
<ul class="list-unstyled">
<li><a href="SUPPORT.html">Getting help</a></li>
</ul>
</div>

<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing DeconvoBuddies</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Lousie Huuki-Myers <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0001-5148-3602" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://codecov.io/gh/lahuuki/DeconvoBuddies?branch=main" class="external-link"><img src="https://codecov.io/gh/lahuuki/DeconvoBuddies/branch/main/graph/badge.svg" alt="Codecov test coverage"></a></li>
<li><a href="https://github.com/lahuuki/DeconvoBuddies/actions" class="external-link"><img src="https://github.com/lahuuki/DeconvoBuddies/workflows/R-CMD-check-bioc/badge.svg" alt="R-CMD-check-bioc"></a></li>
</ul>
</div>

  </div>
</div>


      <footer><div class="copyright">
  <p></p>
<p>Developed by Lousie Huuki-Myers.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
