---
title: "Deconvolution_with_Bisque"
author: 
  - name: Louise Huuki-Myers
    affiliation:
    - &libd Lieber Institute for Brain Development, Johns Hopkins Medical Campus
    email: lahuuki@gmail.com
output: 
  BiocStyle::html_document:
    self_contained: yes
    toc: true
    toc_float: true
    toc_depth: 2
    code_folding: show
date: "`r doc_date()`"
package: "`r pkg_ver('DeconvoBuddies')`"
vignette: >
  %\VignetteIndexEntry{Deconvolution_with_Bisque}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}  
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(
    collapse = TRUE,
    comment = "#>",
    crop = NULL ## Related to https://stat.ethz.ch/pipermail/bioc-devel/2020-April/016656.html
)
```


```{r vignetteSetup, echo=FALSE, message=FALSE, warning = FALSE}
## Track time spent on making the vignette
startTime <- Sys.time()

## Bib setup
library("RefManageR")

## Write bibliography information
bib <- c(
    R = citation(),
    BiocStyle = citation("BiocStyle")[1],
    knitr = citation("knitr")[1],
    RefManageR = citation("RefManageR")[1],
    rmarkdown = citation("rmarkdown")[1],
    sessioninfo = citation("sessioninfo")[1],
    testthat = citation("testthat")[1],
    DeconvoBuddies = citation("DeconvoBuddies")[1]
)
```

#Introduction

## What is Deconvolution?

**Inferring the composition of different cell types in a bulk RNA-seq data**

`Bisque` is an R package for preforming reference based deconvolution [github.com/cozygene/bisque](https://github.com/cozygene/bisque).

## Goals of this Vignette 

We will be demonstrating how to use `DeconvoBuddies` tools when applying 
deconvolution with the `Bisque` package.

1. Install and load required packages 
2. Download and prep example bulk RNA-seq data, and reference snRNA-seq data
3. Find marker genes with `DeconvoBuddies` tools
4. Run deconvolution with `BisqueRNA`
5. Explore deconvolution output and create compostion plots with `DeconvoBuddies` tools


# 1. Install and load required packages

## Install `DeconvoBuddies`

```{r "install", eval = FALSE}
if (!requireNamespace("BiocManager", quietly = TRUE)) {
    install.packages("BiocManager")
}

BiocManager::install("DeconvoBuddies")
BiocManager::install("DeconvoBuddies")

## Check that you have a valid Bioconductor installation
BiocManager::valid()
```


## Load Packages

```{r "load_packages", message=FALSE, warning=FALSE}
## install Bisque from cran
# install.packages("BisqueRNA")

library("spatialLIBD")
library("DeconvoBuddies")
library("SummarizedExperiment")
library("SingleCellExperiment")
library("BisqueRNA")
library("dplyr")
library("tidyr")
library("tibble")
```


# Load Datasets

## Download snRNA-seq data set
Here we download a public single nucleus RNA-seq (snRNA-seq) data from `r Citep(bib[['tran2021']])`
that we'll use as our example. This data can be accessed on [github](https://github.com/LieberInstitute/10xPilot_snRNAseq-human#processed-data).
This data is from postmortem human brain in the dorsolateral prefrontal 
cortex (DLPFC) region, and contains gene expression data for 11k nuclei.

We will use `BiocFileCache()` to cache this data. It is stored as a `SingleCellExperiment` 
object named `sce`, and takes 1.01 GB of RAM memory to load.  

```{r "download spatialDLPFC snRNA-seq"}
sce_path_zip <- fetch_data("spatialDLPFC_snRNAseq")
sce_path <- unzip(sce_path_zip, exdir = tempdir())
sce <- HDF5Array::loadHDF5SummarizedExperiment(
    file.path(tempdir(), "sce_DLPFC_annotated")
)
sce
```


What cell types are in single cell data?
```{r "explore sce"}
## check out the colData of the sce
colData(sce)

## what is the cell type breakdown?
table(sce$cellType_broad_hc)
```

```{r "sn data size check", echo=FALSE}
## Using 172.28 MB
# lobstr::obj_size(sce)
```

## Dowload Bulk DLPFC Dataset
Here we download a public bulk RNA-seq dataset from TODO cite BSP2

This data is from postmortem human brain in the DLPFC Hippocampus (HIPPO), 
and contains gene expression data for 900 samples. We will only use the DLPFC
data in this example. 

We will use `BiocFileCache()` to cache this data. It is stored as a `RangedSummarizedExperiment` 
object named `rse_gene``, and takes 872.12 MB of RAM memory to load.  

```{r "download bulk BSP2 data"}
# Download and save a local cache of the data available at:
# https://github.com/LieberInstitute/10xPilot_snRNAseq-human#processed-data
bfc <- BiocFileCache::BiocFileCache()
url <- paste0(
    "https://s3.us-east-2.amazonaws.com/libd-brainseq2/",
    "rse_gene_unfiltered.Rdata"
)
local_data <- BiocFileCache::bfcrpath(url, x = bfc)
# load rse_gene
load(local_data, verbose = TRUE)
```

```{r "bulk data size check", echo=FALSE}
## Using 872.12 MB
# lobstr::obj_size(rse_gene)
```

```{r "subset bulk data"}
## before subset dimensions
dim(rse_gene)
table(rse_gene$Region)

## subset columns containing samples from the DLPFC
rse_gene <- rse_gene[, rse_gene$Region == "DLPFC"]

## post subset dimensions
dim(rse_gene)
```

# Marker Genes
## Find Marker Genes
```{r, "find marker genes"}
#
# # rm counts to help mem
# gene_sums <- rowSums(as.data.frame(logcounts(sce)))
# top_gene <- gene_sums > median(gene_sums)

## select 1k genes for memory
rownames(sce) <- rowData(sce)$gene_id
sce <- sce[1:1000, ]

marker_stats <- get_mean_ratio2(sce, cellType_col = "cellType_broad_hc")
#
# marker_stats
```

## Plot marker genes
```{r "plot marker genes"}
DeconvoBuddies::plot_marker_express(sce,
    stats = marker_stats,
    cell_type = "Astro",
    cellType_col = "cellType_broad_hc",
    n_genes = 10,
    rank_col = "rank_ratio",
    anno_col = "anno_ratio",
)


# DeconvoBuddies::plot_gene_express(sce,
#                                   cat = "cellType_broad_hc",
#                                   genes = c("MBP","GFAP"))
```

```{r, "pull marker genes"}
#
rownames(rse_gene) <- rowData(rse_gene)$ensemblID

marker_genes <- marker_stats |>
    filter(rank_ratio <= 25, gene %in% rownames(rse_gene)) |>
    pull(gene)

marker_genes <- unique(marker_genes)

length(marker_genes)
```



# Run Bisque

## prep data
```{r "prep data as ExpressionSet"}
exp_set_bulk <- Biobase::ExpressionSet(
    assayData = assays(rse_gene[marker_genes, ])$counts,
    phenoData = AnnotatedDataFrame(
        as.data.frame(colData(rse_gene))[c("SAMPLE_ID")]
    )
)

exp_set_sce <- Biobase::ExpressionSet(
    assayData = as.matrix(assays(sce[marker_genes, ])$counts),
    phenoData = AnnotatedDataFrame(
        as.data.frame(colData(sce))[, c("cellType_broad_hc", "BrNum")]
    )
)

## check for nuclei with 0 marker expression
zero_cell_filter <- colSums(exprs(exp_set_sce)) != 0
message("Exclude ", sum(!zero_cell_filter), " cells")

exp_set_sce <- exp_set_sce[, zero_cell_filter]
```


```{r, "run Bisque"}
est_prop <- ReferenceBasedDecomposition(
    bulk.eset = exp_set_bulk,
    sc.eset = exp_set_sce,
    cell.types = "cellType_broad_hc",
    subject.names = "BrNum",
    use.overlap = FALSE
)
```

# Explore Ouput
```{r, "deconvo output"}
est_prop$bulk.props <- t(est_prop$bulk.props)

head(est_prop$bulk.props)
```

Plot Composition Barplots of Bisque predicted compostions
```{r, "composition plots"}
## add Phenotype data to proportion estimates
pd <- colData(rse_gene) |>
    as.data.frame() |>
    select(Sample = RNum, Sex, Dx)

## make proption estimates long so they are ggplot friendly
prop_long <- est_prop$bulk.props |>
    as.data.frame() |>
    tibble::rownames_to_column("Sample") |>
    tidyr::pivot_longer(!Sample, names_to = "cell_type", values_to = "prop") |>
    left_join(pd)

## create composition bar plots
plot_composition_bar(prop_long = prop_long, sample_col = "Sample", x_col = "Dx")

plot_composition_bar(prop_long = prop_long, sample_col = "Sample", x_col = "Sample", add_text = FALSE)
```


# Reproducibility

The `r Biocpkg("DeconvoBuddies")` package `r Citep(bib[["DeconvoBuddies"]])` was made possible thanks to:

* R `r Citep(bib[["R"]])`
* `r Biocpkg("BiocStyle")` `r Citep(bib[["BiocStyle"]])`
* `r CRANpkg("knitr")` `r Citep(bib[["knitr"]])`
* `r CRANpkg("RefManageR")` `r Citep(bib[["RefManageR"]])`
* `r CRANpkg("rmarkdown")` `r Citep(bib[["rmarkdown"]])`
* `r CRANpkg("sessioninfo")` `r Citep(bib[["sessioninfo"]])`
* `r CRANpkg("testthat")` `r Citep(bib[["testthat"]])`

This package was developed using `r BiocStyle::Biocpkg("biocthis")`.


Code for creating the vignette

```{r createVignette, eval=FALSE}
## Create the vignette
library("rmarkdown")
system.time(render("Deconvolution_with_Bisque.Rmd", "BiocStyle::html_document"))

## Extract the R code
library("knitr")
knit("Deconvolution_with_Bisque.Rmd", tangle = TRUE)
```

Date the vignette was generated.

```{r reproduce1, echo=FALSE}
## Date the vignette was generated
Sys.time()
```

Wallclock time spent generating the vignette.

```{r reproduce2, echo=FALSE}
## Processing time in seconds
totalTime <- diff(c(startTime, Sys.time()))
round(totalTime, digits = 3)
```

`R` session information.

```{r reproduce3, echo=FALSE}
## Session info
library("sessioninfo")
options(width = 120)
session_info()
```



# Bibliography

This vignette was generated using `r Biocpkg("BiocStyle")` `r Citep(bib[["BiocStyle"]])`
with `r CRANpkg("knitr")` `r Citep(bib[["knitr"]])` and `r CRANpkg("rmarkdown")` `r Citep(bib[["rmarkdown"]])` running behind the scenes.

Citations made with `r CRANpkg("RefManageR")` `r Citep(bib[["RefManageR"]])`.

```{r vignetteBiblio, results = "asis", echo = FALSE, warning = FALSE, message = FALSE}
## Print bibliography
PrintBibliography(bib, .opts = list(hyperlink = "to.doc", style = "html"))
```
