<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Calculate 1 vs. All standard fold change for each gene x cell type, wrapper function for scran::findMarkers(). — findMarkers_1vAll • DeconvoBuddies</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Calculate 1 vs. All standard fold change for each gene x cell type, wrapper function for scran::findMarkers(). — findMarkers_1vAll"><meta property="og:description" content='For each cell type, this function computes the statistics comparing that cell
type (the "1") against all other cell types combined ("All").'><meta property="og:image" content="/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">DeconvoBuddies</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/DeconvoBuddies.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/Deconvolution_Benchmark_DLPFC.html">3. Deconvolution Benchmark in Human DLPFC</a>
    </li>
    <li>
      <a href="../articles/Marker_Finding.html">2. Finding Marker Genes with DeconvoBuddies</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/lahuuki/DeconvoBuddies/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Calculate 1 vs. All standard fold change for each gene x cell type, wrapper function for <code>scran::findMarkers()</code>.</h1>
    <small class="dont-index">Source: <a href="https://github.com/lahuuki/DeconvoBuddies/blob/devel/R/findMarkers_1vAll.R" class="external-link"><code>R/findMarkers_1vAll.R</code></a></small>
    <div class="hidden name"><code>findMarkers_1vAll.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>For each cell type, this function computes the statistics comparing that cell
type (the "1") against all other cell types combined ("All").</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">findMarkers_1vAll</span><span class="op">(</span></span>
<span>  <span class="va">sce</span>,</span>
<span>  assay_name <span class="op">=</span> <span class="st">"counts"</span>,</span>
<span>  cellType_col <span class="op">=</span> <span class="st">"cellType"</span>,</span>
<span>  add_symbol <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  mod <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  direction <span class="op">=</span> <span class="st">"up"</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-sce">sce<a class="anchor" aria-label="anchor" href="#arg-sce"></a></dt>
<dd><p>A
<a href="https://rdrr.io/pkg/SingleCellExperiment/man/SingleCellExperiment.html" class="external-link">SingleCellExperiment</a>
object.</p></dd>


<dt id="arg-assay-name">assay_name<a class="anchor" aria-label="anchor" href="#arg-assay-name"></a></dt>
<dd><p>Name of the assay to use for calculation. See
see <code>assayNames(sce)</code> for possible values.</p></dd>


<dt id="arg-celltype-col">cellType_col<a class="anchor" aria-label="anchor" href="#arg-celltype-col"></a></dt>
<dd><p>Column name on <code>colData(sce)</code> that denotes the cell type.</p></dd>


<dt id="arg-add-symbol">add_symbol<a class="anchor" aria-label="anchor" href="#arg-add-symbol"></a></dt>
<dd><p>A <code>logical(1)</code> indicating whether to add the gene symbol
column to the marker stats table.</p></dd>


<dt id="arg-mod">mod<a class="anchor" aria-label="anchor" href="#arg-mod"></a></dt>
<dd><p>A <code>character(1)</code> string specifying the model used as design in
<code><a href="https://rdrr.io/pkg/scran/man/findMarkers.html" class="external-link">scran::findMarkers()</a></code>. It can be <code>NULL</code> (default) if there are no blocking
terms with uninteresting
factors as documented at <a href="https://rdrr.io/pkg/scran/man/pairwiseTTests.html" class="external-link">pairwiseTTests</a>.</p></dd>


<dt id="arg-verbose">verbose<a class="anchor" aria-label="anchor" href="#arg-verbose"></a></dt>
<dd><p>A <code>logical(1)</code> choosing whether to print progress messages or
not.</p></dd>


<dt id="arg-direction">direction<a class="anchor" aria-label="anchor" href="#arg-direction"></a></dt>
<dd><p>A <code>character(1)</code> for the choice of direction tested for
gene cell type markers: <code>"up"</code> (default), <code>"any"</code>, or <code>"down"</code>. Impacts
p-values: if <code>"up"</code> genes with logFC &lt; 0 will have <code>p.value = 1</code>.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>A <code><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble::tibble()</a></code> of 1 vs. ALL standard log fold change + p-values
for each gene x cell type.</p><ul><li><p><code>gene</code> is the name of the gene (from <code>rownames(sce)</code>).</p></li>
<li><p><code>logFC</code> the log fold change from the DE test</p></li>
<li><p><code>log.p.value</code> the log of the p-value of the DE test</p></li>
<li><p><code>log.FDR</code> the log of the False Discovery Rate adjusted p.value</p></li>
<li><p><code>std.logFC</code> the standard logFC.</p></li>
<li><p><code>cellType.target</code> the cell type we're finding marker genes for</p></li>
<li><p><code>std.logFC.rank</code> the rank of <code>std.logFC</code> for each cell type</p></li>
<li><p><code>std.logFC.anno</code> is an annotation of the <code>std.logFC</code> value
helpful for plotting.</p></li>
</ul></div>
    <div id="details">
    <h2>Details</h2>
    <p>See <a href="https://github.com/MarioniLab/scran/issues/57" class="external-link">https://github.com/MarioniLab/scran/issues/57</a> for a more in depth
discussion about the standard log fold change statistics provided by
<code><a href="https://rdrr.io/pkg/scran/man/findMarkers.html" class="external-link">scran::findMarkers()</a></code>.</p>
<p>See also <a href="https://youtu.be/IaclszgZb-g" class="external-link">https://youtu.be/IaclszgZb-g</a> for a LIBD
rstats club presentation on "Finding and interpreting marker genes in
sc/snRNA-seq data". The companion notes are available at
<a href="https://docs.google.com/document/d/1BeMtKgE7gpmNywInndVC9o_ufopn-U2EZHB32bO7ObM/edit?usp=sharing" class="external-link">https://docs.google.com/document/d/1BeMtKgE7gpmNywInndVC9o_ufopn-U2EZHB32bO7ObM/edit?usp=sharing</a>.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co">## load example SingleCellExperiment</span></span></span>
<span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/exists.html" class="external-link">exists</a></span><span class="op">(</span><span class="st">"sce_DLPFC_example"</span><span class="op">)</span><span class="op">)</span> <span class="va">sce_DLPFC_example</span> <span class="op">&lt;-</span> <span class="fu"><a href="fetch_deconvo_data.html">fetch_deconvo_data</a></span><span class="op">(</span><span class="st">"sce_DLPFC_example"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2024-08-23 22:31:10.738115 loading file /github/home/.cache/R/BiocFileCache/1d45c9bd515_sce_DLPFC_example.Rdata%3Frlkey%3Dv3z4u8ru0d2y12zgdl1az07q9%26st%3D1dcfqc1i%26dl%3D1</span>
<span class="r-in"><span><span class="co">## Explore properties of the sce object</span></span></span>
<span class="r-in"><span><span class="va">sce_DLPFC_example</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> class: SingleCellExperiment </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> dim: 557 10000 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> metadata(3): Samples cell_type_colors cell_type_colors_broad</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> assays(1): logcounts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rownames(557): GABRD PRDM16 ... AFF2 MAMLD1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowData names(7): source type ... gene_type binomial_deviance</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colnames(10000): 8_AGTGACTGTAGTTACC-1 17_GCAGCCAGTGAGTCAG-1 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   12_GGACGTCTCTGACAGT-1 1_GGTTAACTCTCTCTAA-1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colData names(32): Sample Barcode ... cellType_layer layer_annotation</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> reducedDimNames(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> mainExpName: NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> altExpNames(0):</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## this data contains logcounts of gene expression</span></span></span>
<span class="r-in"><span><span class="fu">SummarizedExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">sce_DLPFC_example</span><span class="op">)</span><span class="op">$</span><span class="va">logcounts</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           8_AGTGACTGTAGTTACC-1 17_GCAGCCAGTGAGTCAG-1 3_CTGGACGAGCTTCATG-1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GABRD                        0             0.9249246             0.000000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> PRDM16                       0             0.0000000             0.000000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MICOS10                      0             0.0000000             0.000000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> LINC01141                    0             0.0000000             0.000000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ADGRB2                       0             0.9249246             2.253612</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           13_CCCTCAAAGTCTAGCT-1 11_TGTAAGCCATTCTGTT-1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GABRD                  0.000000             0.0000000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> PRDM16                 0.000000             0.0000000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MICOS10                0.000000             0.6528615</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> LINC01141              0.000000             0.0000000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ADGRB2                 2.253454             0.0000000</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## nuclei are classified in to cell types</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">sce_DLPFC_example</span><span class="op">$</span><span class="va">cellType_broad_hc</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     Astro EndoMural     Micro     Oligo       OPC     Excit     Inhib </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       692       417       316      1970       350      4335      1920 </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Get the 1vALL stats for each gene for each cell type defined in</span></span></span>
<span class="r-in"><span><span class="co">## `cellType_broad_hc`</span></span></span>
<span class="r-in"><span><span class="va">marker_stats_1vAll</span> <span class="op">&lt;-</span> <span class="fu">findMarkers_1vAll</span><span class="op">(</span></span></span>
<span class="r-in"><span>    sce <span class="op">=</span> <span class="va">sce_DLPFC_example</span>,</span></span>
<span class="r-in"><span>    assay_name <span class="op">=</span> <span class="st">"logcounts"</span>,</span></span>
<span class="r-in"><span>    cellType_col <span class="op">=</span> <span class="st">"cellType_broad_hc"</span>,</span></span>
<span class="r-in"><span>    mod <span class="op">=</span> <span class="st">"~BrNum"</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running 1vALL Testing for up-regulated genes</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2024-08-23 22:31:11.017756 - Find markers for: Oligo</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2024-08-23 22:31:11.468644 - Find markers for: Excit</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2024-08-23 22:31:11.852284 - Find markers for: Inhib</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2024-08-23 22:31:12.947536 - Find markers for: Micro</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2024-08-23 22:31:13.282228 - Find markers for: OPC</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2024-08-23 22:31:13.618422 - Find markers for: EndoMural</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2024-08-23 22:31:13.954842 - Find markers for: Astro</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2024-08-23 22:31:14.326683 - Building Table</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ** Done! **</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## explore output, top markers have high logFC</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">marker_stats_1vAll</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 6 × 8</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># Groups:   cellType.target [1]</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   gene   logFC log.p.value log.FDR std.logFC cellType.target std.logFC.rank</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>                    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">1</span> ST18    4.46      -<span style="color: #BB0000; text-decoration: underline;">6</span><span style="color: #BB0000;">697.</span>  -<span style="color: #BB0000; text-decoration: underline;">6</span><span style="color: #BB0000;">691.</span>      4.32 Oligo                        1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">2</span> PLP1    4.25      -<span style="color: #BB0000; text-decoration: underline;">5</span><span style="color: #BB0000;">233.</span>  -<span style="color: #BB0000; text-decoration: underline;">5</span><span style="color: #BB0000;">227.</span>      3.50 Oligo                        2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">3</span> MOBP    3.18      -<span style="color: #BB0000; text-decoration: underline;">4</span><span style="color: #BB0000;">821.</span>  -<span style="color: #BB0000; text-decoration: underline;">4</span><span style="color: #BB0000;">816.</span>      3.28 Oligo                        3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">4</span> ENPP2   2.72      -<span style="color: #BB0000; text-decoration: underline;">4</span><span style="color: #BB0000;">521.</span>  -<span style="color: #BB0000; text-decoration: underline;">4</span><span style="color: #BB0000;">516.</span>      3.12 Oligo                        4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">5</span> TF      3.00      -<span style="color: #BB0000; text-decoration: underline;">4</span><span style="color: #BB0000;">466.</span>  -<span style="color: #BB0000; text-decoration: underline;">4</span><span style="color: #BB0000;">462.</span>      3.09 Oligo                        5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">6</span> RNF220  3.80      -<span style="color: #BB0000; text-decoration: underline;">4</span><span style="color: #BB0000;">390.</span>  -<span style="color: #BB0000; text-decoration: underline;">4</span><span style="color: #BB0000;">385.</span>      3.05 Oligo                        6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ 1 more variable: std.logFC.anno &lt;chr&gt;</span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Louise Huuki-Myers.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.9000.</p>
</div>

      </footer></div>






  </body></html>

