<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Get Mean Ratio for Each Gene x Cell Type — get_mean_ratio • DeconvoBuddies</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Get Mean Ratio for Each Gene x Cell Type — get_mean_ratio"><meta property="og:description" content="Calculate the mean ratio value and rank for each gene for each cell type in the sce
object, to identify effective marker genes for deconvolution."><meta property="og:image" content="/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">DeconvoBuddies</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/DeconvoBuddies.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/Deconvolution_Benchmark_DLPFC.html">Deconvolution Benchmark in Human DLPFC</a>
    </li>
    <li>
      <a href="../articles/Marker_Finding.html">Finding Marker Genes with DeconvoBuddies</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/lahuuki/DeconvoBuddies/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Get Mean Ratio for Each Gene x Cell Type</h1>
    <small class="dont-index">Source: <a href="https://github.com/lahuuki/DeconvoBuddies/blob/devel/R/get_mean_ratio.R" class="external-link"><code>R/get_mean_ratio.R</code></a></small>
    <div class="hidden name"><code>get_mean_ratio.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Calculate the mean ratio value and rank for each gene for each cell type in the <code>sce</code>
object, to identify effective marker genes for deconvolution.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">get_mean_ratio</span><span class="op">(</span></span>
<span>  <span class="va">sce</span>,</span>
<span>  <span class="va">cellType_col</span>,</span>
<span>  assay_name <span class="op">=</span> <span class="st">"logcounts"</span>,</span>
<span>  gene_ensembl <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  gene_name <span class="op">=</span> <span class="cn">NULL</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-sce">sce<a class="anchor" aria-label="anchor" href="#arg-sce"></a></dt>
<dd><p><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">SummarizedExperiment-class</a>
(or any derivative class) object containing single cell/nucleus gene expression data</p></dd>


<dt id="arg-celltype-col">cellType_col<a class="anchor" aria-label="anchor" href="#arg-celltype-col"></a></dt>
<dd><p>A <code>character(1)</code> name of the column in the
<a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData()</a> of <code>sce</code> that
denotes the cell type or group of interest.</p></dd>


<dt id="arg-assay-name">assay_name<a class="anchor" aria-label="anchor" href="#arg-assay-name"></a></dt>
<dd><p>A <code>character(1)</code> specifying the name of the
<a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay()</a> in the
<code>sce</code> object to use to rank expression values. Defaults to <code>logcounts</code> since
it typically contains the normalized expression values.</p></dd>


<dt id="arg-gene-ensembl">gene_ensembl<a class="anchor" aria-label="anchor" href="#arg-gene-ensembl"></a></dt>
<dd><p>A <code>character(1)</code> specifying the <code>rowData(sce_pseudo)</code>
column with the ENSEMBL gene IDs. This will be used by <code>layer_stat_cor()</code>.</p></dd>


<dt id="arg-gene-name">gene_name<a class="anchor" aria-label="anchor" href="#arg-gene-name"></a></dt>
<dd><p>A <code>character(1)</code> specifying the <code>rowData(sce_pseudo)</code>
column with the gene names (symbols).</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>A <code>tibble</code> with the <code>MeanRatio</code> values for each gene x cell type.</p><ul><li><p><code>gene</code> is the name of the gene (from rownames(<code>sce</code>)).</p></li>
<li><p><code>cellType.target</code> is the cell type we're finding marker genes for.</p></li>
<li><p><code>mean.target</code> is the mean expression of <code>gene</code> for <code>cellType.target</code>.</p></li>
<li><p><code>cellType.2nd</code> is the second highest non-target cell type.</p></li>
<li><p><code>mean.2nd</code> is the mean expression of <code>gene</code> for <code>cellType.2nd</code>.</p></li>
<li><p><code>MeanRatio</code> is the ratio of <code>mean.target/mean.2nd</code>.</p></li>
<li><p><code>MeanRatio.rank</code> is the rank of <code>MeanRatio</code> for the cell type.</p></li>
<li><p><code>MeanRatio.anno</code> is an annotation of the <code>MeanRatio</code> calculation helpful for plotting.</p></li>
<li><p><code>gene_ensembl</code> &amp; <code>gene_name</code> optional cols spcified by the user to add gene infomation</p></li>
</ul></div>
    <div id="details">
    <h2>Details</h2>
    <p>Note if a cell type has &lt; 10 cells the MeanRatio results may be unstable.
See rational in OSCA: <a href="https://bioconductor.org/books/3.19/OSCA.multisample/multi-sample-comparisons.html#performing-the-de-analysis" class="external-link">https://bioconductor.org/books/3.19/OSCA.multisample/multi-sample-comparisons.html#performing-the-de-analysis</a></p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co">## load example SingleCellExperiment</span></span></span>
<span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/exists.html" class="external-link">exists</a></span><span class="op">(</span><span class="st">"sce_DLPFC_example"</span><span class="op">)</span><span class="op">)</span> <span class="va">sce_DLPFC_example</span> <span class="op">&lt;-</span> <span class="fu"><a href="fetch_deconvo_data.html">fetch_deconvo_data</a></span><span class="op">(</span><span class="st">"sce_DLPFC_example"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2024-08-07 17:33:00.419412 loading file /github/home/.cache/R/BiocFileCache/4a1ec11882_sce_DLPFC_example.Rdata%3Frlkey%3Dv3z4u8ru0d2y12zgdl1az07q9%26st%3D1dcfqc1i%26dl%3D1</span>
<span class="r-in"><span><span class="co">## Explore properties of the sce object</span></span></span>
<span class="r-in"><span><span class="va">sce_DLPFC_example</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> class: SingleCellExperiment </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> dim: 557 10000 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> metadata(3): Samples cell_type_colors cell_type_colors_broad</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> assays(1): logcounts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rownames(557): GABRD PRDM16 ... AFF2 MAMLD1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowData names(7): source type ... gene_type binomial_deviance</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colnames(10000): 8_AGTGACTGTAGTTACC-1 17_GCAGCCAGTGAGTCAG-1 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   12_GGACGTCTCTGACAGT-1 1_GGTTAACTCTCTCTAA-1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colData names(32): Sample Barcode ... cellType_layer layer_annotation</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> reducedDimNames(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> mainExpName: NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> altExpNames(0):</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## this data contains logcounts of gene expression</span></span></span>
<span class="r-in"><span><span class="fu">SummarizedExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">sce_DLPFC_example</span><span class="op">)</span><span class="op">$</span><span class="va">logcounts</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           8_AGTGACTGTAGTTACC-1 17_GCAGCCAGTGAGTCAG-1 3_CTGGACGAGCTTCATG-1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GABRD                        0             0.9249246             0.000000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> PRDM16                       0             0.0000000             0.000000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MICOS10                      0             0.0000000             0.000000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> LINC01141                    0             0.0000000             0.000000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ADGRB2                       0             0.9249246             2.253612</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           13_CCCTCAAAGTCTAGCT-1 11_TGTAAGCCATTCTGTT-1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GABRD                  0.000000             0.0000000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> PRDM16                 0.000000             0.0000000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MICOS10                0.000000             0.6528615</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> LINC01141              0.000000             0.0000000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ADGRB2                 2.253454             0.0000000</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## nuclei are classified in to cell types</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">sce_DLPFC_example</span><span class="op">$</span><span class="va">cellType_broad_hc</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     Astro EndoMural     Micro     Oligo       OPC     Excit     Inhib </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       692       417       316      1970       350      4335      1920 </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Get the mean ratio for each gene for each cell type defined in `cellType_broad_hc`</span></span></span>
<span class="r-in"><span><span class="fu">get_mean_ratio</span><span class="op">(</span><span class="va">sce_DLPFC_example</span>, cellType_col <span class="op">=</span> <span class="st">"cellType_broad_hc"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 762 × 8</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    gene       cellType.target mean.target cellType.2nd mean.2nd MeanRatio</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>                 <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 1</span> CD22       Oligo                  1.36 OPC            0.073<span style="text-decoration: underline;">0</span>      18.6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 2</span> LINC01608  Oligo                  2.39 Micro          0.142       16.8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 3</span> FOLH1      Oligo                  1.59 OPC            0.101       15.7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 4</span> SLC5A11    Oligo                  2.14 Micro          0.145       14.7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 5</span> AC012494.1 Oligo                  2.42 OPC            0.169       14.3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 6</span> ST18       Oligo                  4.65 OPC            0.329       14.1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 7</span> MAG        Oligo                  1.44 Astro          0.103       14.0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 8</span> ANLN       Oligo                  1.60 Micro          0.115       13.9</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 9</span> CLDN11     Oligo                  1.82 EndoMural      0.146       12.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">10</span> MOG        Oligo                  2.06 OPC            0.185       11.1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ 752 more rows</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ 2 more variables: MeanRatio.rank &lt;int&gt;, MeanRatio.anno &lt;chr&gt;</span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Option to specify gene_name as the "Symbol" column from rowData</span></span></span>
<span class="r-in"><span><span class="co"># this will be added to the marker stats output</span></span></span>
<span class="r-in"><span><span class="fu">SummarizedExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">sce_DLPFC_example</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> DataFrame with 557 rows and 7 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             source     type         gene_id gene_version   gene_name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           &lt;factor&gt; &lt;factor&gt;     &lt;character&gt;  &lt;character&gt; &lt;character&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GABRD       HAVANA     gene ENSG00000187730            9       GABRD</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> PRDM16      HAVANA     gene ENSG00000142611           17      PRDM16</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MICOS10     HAVANA     gene ENSG00000173436           15     MICOS10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> LINC01141   HAVANA     gene ENSG00000236963            7   LINC01141</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ADGRB2      HAVANA     gene ENSG00000121753           12      ADGRB2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ...            ...      ...             ...          ...         ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> TRPC5       HAVANA     gene ENSG00000072315            3       TRPC5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> LAMP2       HAVANA     gene ENSG00000005893           15       LAMP2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> RTL8C       HAVANA     gene ENSG00000134590           14       RTL8C</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> AFF2        HAVANA     gene ENSG00000155966           14        AFF2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MAMLD1      HAVANA     gene ENSG00000013619           14      MAMLD1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                gene_type binomial_deviance</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>              &lt;character&gt;         &lt;numeric&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GABRD     protein_coding           69168.8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> PRDM16    protein_coding           81602.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MICOS10   protein_coding           96788.7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> LINC01141         lncRNA           35228.6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ADGRB2    protein_coding           81087.8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ...                  ...               ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> TRPC5     protein_coding          134934.0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> LAMP2     protein_coding          132756.1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> RTL8C     protein_coding           98554.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> AFF2      protein_coding          111683.6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MAMLD1    protein_coding           75492.1</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## specify rowData col names for gene_name and gene_ensembl</span></span></span>
<span class="r-in"><span><span class="fu">get_mean_ratio</span><span class="op">(</span><span class="va">sce_DLPFC_example</span>, cellType_col <span class="op">=</span> <span class="st">"cellType_broad_hc"</span>, gene_name <span class="op">=</span> <span class="st">"gene_name"</span>, gene_ensembl <span class="op">=</span> <span class="st">"gene_id"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 762 × 10</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    gene       cellType.target mean.target cellType.2nd mean.2nd MeanRatio</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>                 <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 1</span> CD22       Oligo                  1.36 OPC            0.073<span style="text-decoration: underline;">0</span>      18.6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 2</span> LINC01608  Oligo                  2.39 Micro          0.142       16.8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 3</span> FOLH1      Oligo                  1.59 OPC            0.101       15.7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 4</span> SLC5A11    Oligo                  2.14 Micro          0.145       14.7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 5</span> AC012494.1 Oligo                  2.42 OPC            0.169       14.3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 6</span> ST18       Oligo                  4.65 OPC            0.329       14.1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 7</span> MAG        Oligo                  1.44 Astro          0.103       14.0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 8</span> ANLN       Oligo                  1.60 Micro          0.115       13.9</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 9</span> CLDN11     Oligo                  1.82 EndoMural      0.146       12.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">10</span> MOG        Oligo                  2.06 OPC            0.185       11.1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ 752 more rows</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ 4 more variables: MeanRatio.rank &lt;int&gt;, MeanRatio.anno &lt;chr&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   gene_ensembl &lt;chr&gt;, gene_name &lt;chr&gt;</span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Lousie Huuki-Myers.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.9000.</p>
</div>

      </footer></div>






  </body></html>

