<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>3. Deconvolution Benchmark in Human DLPFC • DeconvoBuddies</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="3. Deconvolution Benchmark in Human DLPFC">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">DeconvoBuddies</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/DeconvoBuddies.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Deconvolution_Benchmark_DLPFC.html">3. Deconvolution Benchmark in Human DLPFC</a>
    </li>
    <li>
      <a href="../articles/Marker_Finding.html">2. Finding Marker Genes with DeconvoBuddies</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/lahuuki/DeconvoBuddies/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>3. Deconvolution Benchmark in Human DLPFC</h1>
                        <h4 data-toc-skip class="author">Louise
Huuki-Myers</h4>
            <address class="author_afil">
      Lieber Institute for Brain Development, Johns Hopkins Medical
Campus<br><a class="author_email" href="mailto:#"></a><a href="mailto:lahuuki@gmail.com" class="email">lahuuki@gmail.com</a>
      </address>
                  
            <h4 data-toc-skip class="date">23 August 2024</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/lahuuki/DeconvoBuddies/blob/devel/vignettes/Deconvolution_Benchmark_DLPFC.Rmd" class="external-link"><code>vignettes/Deconvolution_Benchmark_DLPFC.Rmd</code></a></small>
      <div class="hidden name"><code>Deconvolution_Benchmark_DLPFC.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<div class="section level3">
<h3 id="what-is-deconvolution">What is Deconvolution?<a class="anchor" aria-label="anchor" href="#what-is-deconvolution"></a>
</h3>
<p><strong>Inferring the composition of different cell types in a bulk
RNA-seq data</strong></p>
<p>Deconvolution is a analysis that aims to calculate the proportion of
different cell types that make up a sample of bulk RNA-seq, based off of
cell type gene expression profiles in a single cell/nuclei RNA-seq
dataset.</p>
<p align="center">
</p>
<div class="float">
<img src="http://research.libd.org/DeconvoBuddies/reference/figures/Deconvolution.png" alt="Figure 1. Overview on how to use single cell data to infer the composition of bulk RNA-seq samples"><div class="figcaption">Figure 1. Overview on how to use single cell
data to infer the composition of bulk RNA-seq samples</div>
</div>

<div class="section level4">
<h4 id="deconvolution-methods">Deconvolution Methods<a class="anchor" aria-label="anchor" href="#deconvolution-methods"></a>
</h4>
<p>There are 20+ published reference based deconvolution methods. Below
are a selection of 6 methods we tested in our <a href="https://doi.org/10.1101/2024.02.09.579665" class="external-link">deconvolution benchmark
study</a>.</p>
<table class="table">
<colgroup>
<col width="25%">
<col width="22%">
<col width="22%">
<col width="28%">
</colgroup>
<thead><tr class="header">
<th>Approach</th>
<th>Method</th>
<th>Citation</th>
<th>Availability</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>weighted least squares</td>
<td>DWLS</td>
<td>Tsoucas et al, Nature Comm, 2019</td>
<td><a href="https://cran.r-project.org/web/packages/DWLS/index.html" class="external-link">R
Package CRAN</a></td>
</tr>
<tr class="even">
<td>Bias correction: Assay</td>
<td>Bisque</td>
<td>Jew et al, Nature Comm, 2020</td>
<td><a href="https://github.com/cozygene/bisque" class="external-link">R Package
GitHub</a></td>
</tr>
<tr class="odd">
<td>Bias correction: Source</td>
<td>MuSiC</td>
<td>Wang et al, Nature Communications, 2019</td>
<td><a href="https://github.com/xuranw/MuSiC" class="external-link">R Package GitHub</a></td>
</tr>
<tr class="even">
<td>Machine Learning</td>
<td>CIBERSORTx</td>
<td>Newman et al., Nature BioTech, 2019</td>
<td><a href="https://cibersortx.stanford.edu/" class="external-link">Webtool</a></td>
</tr>
<tr class="odd">
<td>Bayesian</td>
<td>BayesPrism</td>
<td>Chu et al., Nature Cancer, 2022</td>
<td><a href="https://github.com/Danko-Lab/BayesPrism" class="external-link">Webtool/R
Package</a></td>
</tr>
<tr class="even">
<td>linear</td>
<td>Hspe</td>
<td>Hunt et al., Ann. Appl. Stat, 2021</td>
<td><a href="https://gjhunt.github.io/hspe/" class="external-link">R package GitHub</a></td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="section level3">
<h3 id="goals-of-this-vignette">Goals of this Vignette<a class="anchor" aria-label="anchor" href="#goals-of-this-vignette"></a>
</h3>
<p>We will be demonstrating how to use <code>DeconvoBuddies</code> tools
when applying deconvolution with the <code>Bisque</code> package.</p>
<ol style="list-style-type: decimal">
<li>Install and load required packages</li>
<li>Download DLPFC RNA-seq data, and reference snRNA-seq data</li>
<li>Find marker genes with <code>DeconvoBuddies</code> tools</li>
<li>Run deconvolution with <code>BisqueRNA</code>
</li>
<li>Explore deconvolution output and create composition plots with
<code>DeconvoBuddies</code> tools</li>
<li>Check proportion against RNAScope estimated proportions</li>
</ol>
</div>
<div class="section level3">
<h3 id="video-tutorial">Video Tutorial<a class="anchor" aria-label="anchor" href="#video-tutorial"></a>
</h3>
<p>Linked is a video from a presentation of an earlier version of this
tutorial from our <a href="https://research.libd.org/rstatsclub/" class="external-link">LIBD
Rstats club</a>.</p>
<iframe width="560" height="315" src="https://www.youtube.com/embed/6fTnp_hcYeI?si=UbkGNclU2q8I8WXO" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" referrerpolicy="strict-origin-when-cross-origin" allowfullscreen>
</iframe>
</div>
</div>
<div class="section level2">
<h2 id="basics">Basics<a class="anchor" aria-label="anchor" href="#basics"></a>
</h2>
</div>
<div class="section level2">
<h2 id="install-and-load-required-packages">1. Install and load required packages<a class="anchor" aria-label="anchor" href="#install-and-load-required-packages"></a>
</h2>
<p><code>R</code> is an open-source statistical environment which can be
easily modified to enhance its functionality via packages. <em><a href="https://bioconductor.org/packages/3.20/DeconvoBuddies" class="external-link">DeconvoBuddies</a></em>
is a <code>R</code> package available via the <a href="http://bioconductor.org" class="external-link">Bioconductor</a> repository for packages.
<code>R</code> can be installed on any operating system from <a href="https://cran.r-project.org/" class="external-link">CRAN</a> after which you can install
<em><a href="https://bioconductor.org/packages/3.20/DeconvoBuddies" class="external-link">DeconvoBuddies</a></em>
by using the following commands in your <code>R</code> session:</p>
<div class="section level3">
<h3 id="install-deconvobuddies">Install <code>DeconvoBuddies</code><a class="anchor" aria-label="anchor" href="#install-deconvobuddies"></a>
</h3>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"DeconvoBuddies"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Check that you have a valid Bioconductor installation</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/valid.html" class="external-link">valid</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="load-other-packages">Load Other Packages<a class="anchor" aria-label="anchor" href="#load-other-packages"></a>
</h3>
<p>Let’s load the packages we’ll be using in this vignette.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Packages for different types of RNA-seq data structures in R</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.org/packages/SummarizedExperiment" class="external-link">"SummarizedExperiment"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"SingleCellExperiment"</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.org/packages/Biobase" class="external-link">"Biobase"</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co">## For downloading data</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/LieberInstitute/spatialLIBD" class="external-link">"spatialLIBD"</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co">## For running deconvolution</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://www.biorxiv.org/content/10.1101/669911v1" class="external-link">"BisqueRNA"</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Other helper packages for this vignette</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://dplyr.tidyverse.org" class="external-link">"dplyr"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://tidyr.tidyverse.org" class="external-link">"tidyr"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://tibble.tidyverse.org/" class="external-link">"tibble"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://ggplot2.tidyverse.org" class="external-link">"ggplot2"</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Our main package</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/lahuuki/DeconvoBuddies" class="external-link">"DeconvoBuddies"</a></span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="download-dlpfc-rna-seq-data-and-reference-snrna-seq-data-">2. Download DLPFC RNA-seq data, and reference snRNA-seq data.<a class="anchor" aria-label="anchor" href="#download-dlpfc-rna-seq-data-and-reference-snrna-seq-data-"></a>
</h2>
<div class="section level3">
<h3 id="bulk-rna-seq-data">Bulk RNA-seq data<a class="anchor" aria-label="anchor" href="#bulk-rna-seq-data"></a>
</h3>
<p>Access the 110 sample Human DLPFC bulk RNA-seq dataset for LIBD
described in more detail <a href="https://github.com/LieberInstitute/Human_DLPFC_Deconvolution" class="external-link">here</a>.
These samples<br>
are from 19 tissue blocks, and 10 neurotypical adult donors. Samples
were sequenced with two different <code>library_types</code> (polyA and
RiboZeroGold), and three different <code>RNA_extraction</code> (Cyto,
Total, Nuc). There are in total n=110 samples after quality control.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## use fetch deconvo data to load rse_gene</span></span>
<span><span class="va">rse_gene</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/fetch_deconvo_data.html">fetch_deconvo_data</a></span><span class="op">(</span><span class="st">"rse_gene"</span><span class="op">)</span></span>
<span><span class="co">#&gt; 2024-08-23 22:32:25.907382 loading file /github/home/.cache/R/BiocFileCache/1d419ca4697_rse_gene.Rdata%3Frlkey%3Dsw2djr71y954yw4o3xrmjv59b%26dl%3D1</span></span>
<span><span class="va">rse_gene</span></span>
<span><span class="co">#&gt; class: RangedSummarizedExperiment </span></span>
<span><span class="co">#&gt; dim: 21745 110 </span></span>
<span><span class="co">#&gt; metadata(1): SPEAQeasy_settings</span></span>
<span><span class="co">#&gt; assays(2): counts logcounts</span></span>
<span><span class="co">#&gt; rownames(21745): ENSG00000227232.5 ENSG00000278267.1 ...</span></span>
<span><span class="co">#&gt;   ENSG00000210195.2 ENSG00000210196.2</span></span>
<span><span class="co">#&gt; rowData names(11): Length gencodeID ... gencodeTx passExprsCut</span></span>
<span><span class="co">#&gt; colnames(110): 2107UNHS-0291_Br2720_Mid_Bulk</span></span>
<span><span class="co">#&gt;   2107UNHS-0291_Br2720_Mid_Cyto ... AN00000906_Br8667_Mid_Cyto</span></span>
<span><span class="co">#&gt;   AN00000906_Br8667_Mid_Nuc</span></span>
<span><span class="co">#&gt; colData names(80): SAMPLE_ID Sample ... diagnosis qc_class</span></span>
<span><span class="co"># lobstr::obj_size(rse_gene)</span></span>
<span><span class="co"># 41.16 MB</span></span>
<span></span>
<span><span class="co">## Use gene "Symbol" as identifiers for the genes in rownames(rse_gene)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span><span class="op">$</span><span class="va">Symbol</span></span>
<span></span>
<span><span class="co">## bulk RNA seq samples were sequenced with different library types,</span></span>
<span><span class="co">## and RNA extractions</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">$</span><span class="va">library_type</span>, <span class="va">rse_gene</span><span class="op">$</span><span class="va">library_prep</span><span class="op">)</span></span>
<span><span class="co">#&gt;               </span></span>
<span><span class="co">#&gt;                Bulk Cyto Nuc</span></span>
<span><span class="co">#&gt;   polyA          19   18  18</span></span>
<span><span class="co">#&gt;   RiboZeroGold   19   19  17</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="reference-snrna-seq-data">Reference snRNA-seq data<a class="anchor" aria-label="anchor" href="#reference-snrna-seq-data"></a>
</h3>
<p>This data is paired with a single nucleus RNA-seq data set from
<code>spatialLIBD</code>. This dataset can be accessed with
<code><a href="https://rdrr.io/pkg/spatialLIBD/man/fetch_data.html" class="external-link">spatialLIBD::fetch_data()</a></code>.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Use spatialLIBD to fetch the snRNA-seq dataset used in this project</span></span>
<span><span class="va">sce_path_zip</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/fetch_deconvo_data.html">fetch_deconvo_data</a></span><span class="op">(</span><span class="st">"sce"</span><span class="op">)</span></span>
<span><span class="co">#&gt; 2024-08-23 22:32:29.043671 loading file /github/home/.cache/R/BiocFileCache/1d4602f9741_sce_DLPFC_annotated.zip%3Fdl%3D1</span></span>
<span></span>
<span><span class="co">## unzip and load the data</span></span>
<span><span class="va">sce_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/unzip.html" class="external-link">unzip</a></span><span class="op">(</span><span class="va">sce_path_zip</span>, exdir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu">HDF5Array</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/HDF5Array/man/saveHDF5SummarizedExperiment.html" class="external-link">loadHDF5SummarizedExperiment</a></span><span class="op">(</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>, <span class="st">"sce_DLPFC_annotated"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># lobstr::obj_size(sce)</span></span>
<span><span class="co"># 172.28 MB</span></span>
<span></span>
<span><span class="co">## exclude Ambiguous cell type</span></span>
<span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="va">sce</span><span class="op">[</span>, <span class="va">sce</span><span class="op">$</span><span class="va">cellType_broad_hc</span> <span class="op">!=</span> <span class="st">"Ambiguous"</span><span class="op">]</span></span>
<span><span class="va">sce</span><span class="op">$</span><span class="va">cellType_broad_hc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/droplevels.html" class="external-link">droplevels</a></span><span class="op">(</span><span class="va">sce</span><span class="op">$</span><span class="va">cellType_broad_hc</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Check the number of genes by the number of nuclei that we</span></span>
<span><span class="co">## have to work with:</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 36601 56447</span></span>
<span></span>
<span><span class="co">## Check the broad cell type distribution</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">sce</span><span class="op">$</span><span class="va">cellType_broad_hc</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     Astro EndoMural     Micro     Oligo       OPC     Excit     Inhib </span></span>
<span><span class="co">#&gt;      3979      2157      1601     10894      1940     24809     11067</span></span>
<span></span>
<span><span class="co">## We're going to subset to the first 5k genes to save memory</span></span>
<span><span class="co">## just for this example. You wouldn't do this on a full analysis.</span></span>
<span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="va">sce</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fl">5000</span><span class="op">)</span>, <span class="op">]</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="orthogonal-cell-type-proportion-from-rnascopeif">Orthogonal Cell Type Proportion from RNAScope/IF<a class="anchor" aria-label="anchor" href="#orthogonal-cell-type-proportion-from-rnascopeif"></a>
</h3>
<p>An alternative method for calculating cell type proportions is
through imaging a slice of tissue with cell type probes using single
molecule fluorescent <em>in situ</em> hybridization (smFISH) experiments
performed with RNAScope/IF (ImmunoFluorescence). Then analyse the image
to count cells and annotate them by cell type. In this study we used <a href="https://indicalab.com/halo/" class="external-link">HALO from Indica Labs</a> for this
step.</p>
<p>The cell type proportions from the RNAScope/IF experiment will be
used to evaluate the accuracy of cell type proportion estimates.</p>
<p align="center">
</p>
<div class="float">
<img src="http://research.libd.org/DeconvoBuddies/reference/figures/Deconvolution_compare_proportions.png" alt="Figure 2. RNAScope/IF measures the cell type proportions through imaging"><div class="figcaption">Figure 2. RNAScope/IF measures the cell type
proportions through imaging</div>
</div>

<p>The RNAScope/IF proportion data is stored as a
<code>data.frame</code> object in
<code><a href="../reference/RNAScope_prop.html">DeconvoBuddies::RNAScope_prop</a></code>.</p>
<p>Key columns in <code>RNAScope_prop</code>:</p>
<ul>
<li><p><code>SAMPLE_ID</code>: DLPFC Tissue block + RNAScope
combination.</p></li>
<li><p><code>Sample</code> : DLFPC Tissue block (Donor BrNum + DLPFC
position).</p></li>
<li><p><code>cell_type</code> : The cell type measured.</p></li>
<li><p><code>n_cell</code> : the number of cells counted for the Sample
and cell type.</p></li>
<li><p><code>prop</code> : the calculated cell type proportion from
n_cell</p></li>
</ul>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Access the RNAScope proportion data.frame</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu">DeconvoBuddies</span><span class="fu">::</span><span class="va"><a href="../reference/RNAScope_prop.html">RNAScope_prop</a></span><span class="op">)</span></span>
<span><span class="co">#&gt;        SAMPLE_ID      Sample  Combo cell_type Confidence n_cell       prop</span></span>
<span><span class="co">#&gt; 1 Br2720P_CIRCLE Br2720_post Circle     Astro       High   3385 0.09357808</span></span>
<span><span class="co">#&gt; 2 Br2720P_CIRCLE Br2720_post Circle EndoMural       High   1680 0.04644348</span></span>
<span><span class="co">#&gt; 3 Br2720P_CIRCLE Br2720_post Circle     Inhib       High   3092 0.08547812</span></span>
<span><span class="co">#&gt; 4 Br2720P_CIRCLE Br2720_post Circle     Other       High  28016 0.77450032</span></span>
<span><span class="co">#&gt; 5   Br2720P_STAR Br2720_post   Star     Excit       High   8687 0.23132639</span></span>
<span><span class="co">#&gt; 6   Br2720P_STAR Br2720_post   Star     Micro       High   1416 0.03770671</span></span>
<span><span class="co">#&gt;   n_cell_sn    prop_sn</span></span>
<span><span class="co">#&gt; 1        81 0.04287983</span></span>
<span><span class="co">#&gt; 2        66 0.03493912</span></span>
<span><span class="co">#&gt; 3        94 0.04976178</span></span>
<span><span class="co">#&gt; 4        NA         NA</span></span>
<span><span class="co">#&gt; 5      1198 0.63419799</span></span>
<span><span class="co">#&gt; 6        47 0.02488089</span></span>
<span></span>
<span><span class="co">## plot the RNAScope compositions</span></span>
<span><span class="fu"><a href="../reference/plot_composition_bar.html">plot_composition_bar</a></span><span class="op">(</span></span>
<span>    prop_long <span class="op">=</span> <span class="va">RNAScope_prop</span>,</span>
<span>    sample_col <span class="op">=</span> <span class="st">"SAMPLE_ID"</span>,</span>
<span>    x_col <span class="op">=</span> <span class="st">"SAMPLE_ID"</span>,</span>
<span>    add_text <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">Combo</span>, nrow <span class="op">=</span> <span class="fl">2</span>, scales <span class="op">=</span> <span class="st">"free_x"</span><span class="op">)</span></span></code></pre></div>
<p><img src="Deconvolution_Benchmark_DLPFC_files/figure-html/access%20RNAScope%20proportions-1.png" width="700"></p>
<p>Above we can see the cell proportions for each of the samples in
either the <code>Circle</code> or the <code>Star</code> combination of
RNAScope/IF probes. Each combination of RNAScope/IF probes was able to
assess different sets of cell types. Two of them had to be used to
measure the number of cell types studied in this case as there are
limits on how many unique probes you can measure in a given RNAScope/IF
experiment. For more details about the RNAScope/IF combinations, check
the paper describing <a href="https://github.com/LieberInstitute/Human_DLPFC_Deconvolution" class="external-link">this
study</a>.</p>
</div>
</div>
<div class="section level2">
<h2 id="select-marker-genes">3. Select Marker Genes<a class="anchor" aria-label="anchor" href="#select-marker-genes"></a>
</h2>
<p>Marker genes are genes with high expression in one cell type and low
expression in other cell types, or “cell-type specific” expression.
These genes can be used to learn more about the identity and function of
cell types, but here we are interested in using a sets of cell type
specific marker genes to reduce noise in deconvolution and increase
accuracy.</p>
<p>We have developed a method for finding marker genes called the “Mean
Ratio”. We calculate the <em>MeanRatio</em> for a target cell type for
each gene by dividing the mean expression of the target cell by the mean
expression of the next highest non-target cell type. Genes with the
highest <em>MeanRatio</em> values are selected as marker genes.</p>
<p>For a tutorial on marker gene selection check out <a href="https://research.libd.org/DeconvoBuddies/articles/Marker_Finding.html" class="external-link">Vignette:
Finding Marker Genes with Deconvo Buddies</a>.</p>
<p align="center">
</p>
<div class="float">
<img src="http://research.libd.org/DeconvoBuddies/reference/figures/get_mean_ratio.png" alt="Figure 3. Mean Ratio calculation process compared to 1vALL Marker Gene selection"><div class="figcaption">Figure 3. Mean Ratio calculation process
compared to 1vALL Marker Gene selection</div>
</div>

<div class="section level3">
<h3 id="use-get_mean_ratio-to-find-marker-genes-">Use <code>get_mean_ratio()</code> to find marker genes.<a class="anchor" aria-label="anchor" href="#use-get_mean_ratio-to-find-marker-genes-"></a>
</h3>
<p>The function <code><a href="../reference/get_mean_ratio.html">DeconvoBuddies::get_mean_ratio()</a></code> calculates
the <em>MeanRatio</em> and the rank of genes for a specified cell type
annotation in an <code>SingleCellExperiment</code> object.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># calculate the Mean Ratio of genes for each cell type</span></span>
<span><span class="va">marker_stats</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_mean_ratio.html">get_mean_ratio</a></span><span class="op">(</span><span class="va">sce</span>,</span>
<span>    cellType_col <span class="op">=</span> <span class="st">"cellType_broad_hc"</span>,</span>
<span>    gene_ensembl <span class="op">=</span> <span class="st">"gene_id"</span>,</span>
<span>    gene_name <span class="op">=</span> <span class="st">"gene_name"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># check the top gene ranked gene for each cell type</span></span>
<span><span class="va">marker_stats</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">cellType.target</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/slice.html" class="external-link">slice</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 7 × 10</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># Groups:   cellType.target [7]</span></span></span>
<span><span class="co">#&gt;   gene       cellType.target mean.target cellType.2nd mean.2nd MeanRatio</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>                 <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> PRDM16     Astro                  1.97 EndoMural      0.142      13.9 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> SLC2A1     EndoMural              1.49 Astro          0.146      10.2 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> LINC01141  Micro                  1.57 Excit          0.064<span style="text-decoration: underline;">0</span>     24.5 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> AC012494.1 Oligo                  2.37 OPC            0.147      16.1 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> MIR3681HG  OPC                    1.58 Excit          0.205       7.69</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> AC011995.2 Excit                  1.01 Inhib          0.135       7.51</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">7</span> LYPD6B     Inhib                  1.20 Excit          0.096<span style="text-decoration: underline;">7</span>     12.4 </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 4 more variables: MeanRatio.rank &lt;int&gt;, MeanRatio.anno &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   gene_ensembl &lt;chr&gt;, gene_name &lt;chr&gt;</span></span></span></code></pre></div>
<p>The columns of this table are documented in detail at
<code><a href="../reference/get_mean_ratio.html">DeconvoBuddies::get_mean_ratio()</a></code>. Though
<code>cellType.target</code> lists the target cell type for which the
<em>MeanRatio</em> is being calculated. It is the numerator of the
<em>MeanRatio</em>. The <code>cellType.2nd</code> lists the cell type
against which the target cell type is being compared. It is the
denominator of the <em>MeanRatio</em>.</p>
</div>
<div class="section level3">
<h3 id="plot-the-top-marker-genes">Plot the top marker genes<a class="anchor" aria-label="anchor" href="#plot-the-top-marker-genes"></a>
</h3>
<p>Use <code>DeconvoBuddies</code> plotting tools to quickly plot the
gene expression of the top 4 Excitatory neuron marker genes across the
<code>cellType_broad_hc</code> cell type annotations.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># plot expression across cell type the top 4 Excit marker genes</span></span>
<span><span class="fu"><a href="../reference/plot_marker_express.html">plot_marker_express</a></span><span class="op">(</span><span class="va">sce</span>,</span>
<span>    stats <span class="op">=</span> <span class="va">marker_stats</span>,</span>
<span>    cell_type <span class="op">=</span> <span class="st">"Excit"</span>,</span>
<span>    cellType_col <span class="op">=</span> <span class="st">"cellType_broad_hc"</span>,</span>
<span>    rank_col <span class="op">=</span> <span class="st">"MeanRatio.rank"</span>,</span>
<span>    anno_col <span class="op">=</span> <span class="st">"MeanRatio.anno"</span>,</span>
<span>    gene_col <span class="op">=</span> <span class="st">"gene"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="Deconvolution_Benchmark_DLPFC_files/figure-html/plot%20marker%20genes-1.png" width="700"></p>
<p>Looks nice and cell type specific!</p>
<p>Note how <code><a href="../reference/plot_marker_express.html">plot_marker_express()</a></code> lists the 2 cell types
that are being compared that result in the specific numerical
<em>MeanRatio</em> value being displayed. As we specified that our
target cell type is <code>"Excit"</code> in our call to
<code><a href="../reference/plot_marker_express.html">plot_marker_express()</a></code>, the numerator is <code>Excit</code>
in all the panels shown above.</p>
<div class="section level4">
<h4 id="create-a-list-of-marker-genes">Create a List of Marker Genes<a class="anchor" aria-label="anchor" href="#create-a-list-of-marker-genes"></a>
</h4>
<p>With the <em>MeanRatio</em> calculated, we will select the top 25
highest Mean Ratio genes for each cell type, that also exists in the
bulk data <code>rse_gene</code>.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># select top 25 marker genes for each cell type, that are also in rse_gene</span></span>
<span><span class="va">marker_genes</span> <span class="op">&lt;-</span> <span class="va">marker_stats</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">MeanRatio.rank</span> <span class="op">&lt;=</span> <span class="fl">25</span> <span class="op">&amp;</span> <span class="va">gene</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># check how many genes for each cell type (some genes are not in both datasets)</span></span>
<span><span class="va">marker_genes</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html" class="external-link">count</a></span><span class="op">(</span><span class="va">cellType.target</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 7 × 2</span></span></span>
<span><span class="co">#&gt;   cellType.target     n</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> Astro              22</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> EndoMural          25</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> Micro              24</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> Oligo              23</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> OPC                22</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> Excit              21</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">7</span> Inhib              24</span></span>
<span></span>
<span><span class="co"># create a vector of marker genes to subset data before deconvolution</span></span>
<span><span class="va">marker_genes</span> <span class="op">&lt;-</span> <span class="va">marker_genes</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">gene</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="prep-data-and-run-bisque">4. Prep Data and Run Bisque<a class="anchor" aria-label="anchor" href="#prep-data-and-run-bisque"></a>
</h2>
<p><a href="https://github.com/cozygene/bisque" class="external-link"><em>Bisque</em></a> is
an R package for cell type deconvolution. In our <a href="https://github.com/LieberInstitute/Human_DLPFC_Deconvolution" class="external-link">deconvolution
benchmark</a>, we found it was a top performing method. Below we will
briefly show how to run <code>Bisque</code>’s “reference based
decomposition (deconvolution)”.</p>
<div class="section level3">
<h3 id="prepare-data">Prepare data<a class="anchor" aria-label="anchor" href="#prepare-data"></a>
</h3>
<p>To run <code>Bisque</code> the snRNA-seq and bulk data must first be
converted to <code><a href="https://rdrr.io/pkg/Biobase/man/class.ExpressionSet.html" class="external-link">Biobase::ExpressionSet()</a></code> format. We will
subset our data to our selected <em>MeanRatio</em> marker genes.</p>
<p>The snRNA-seq data must also be filtered for cells with no counts
across marker genes.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## convert bulk data to Expression set, sub-setting to marker genes</span></span>
<span><span class="co">## include sample ID</span></span>
<span><span class="va">exp_set_bulk</span> <span class="op">&lt;-</span> <span class="fu">Biobase</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/class.ExpressionSet.html" class="external-link">ExpressionSet</a></span><span class="op">(</span></span>
<span>    assayData <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">[</span><span class="va">marker_genes</span>, <span class="op">]</span><span class="op">)</span><span class="op">$</span><span class="va">counts</span>,</span>
<span>    phenoData <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/class.AnnotatedDataFrame.html" class="external-link">AnnotatedDataFrame</a></span><span class="op">(</span></span>
<span>        <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"SAMPLE_ID"</span><span class="op">)</span><span class="op">]</span></span>
<span>    <span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## convert snRNA-seq data to Expression set, sub-setting to marker genes</span></span>
<span><span class="co">## include cell type and donor information</span></span>
<span><span class="va">exp_set_sce</span> <span class="op">&lt;-</span> <span class="fu">Biobase</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/class.ExpressionSet.html" class="external-link">ExpressionSet</a></span><span class="op">(</span></span>
<span>    assayData <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">sce</span><span class="op">[</span><span class="va">marker_genes</span>, <span class="op">]</span><span class="op">)</span><span class="op">$</span><span class="va">counts</span><span class="op">)</span>,</span>
<span>    phenoData <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/class.AnnotatedDataFrame.html" class="external-link">AnnotatedDataFrame</a></span><span class="op">(</span></span>
<span>        <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">)</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"cellType_broad_hc"</span>, <span class="st">"BrNum"</span><span class="op">)</span><span class="op">]</span></span>
<span>    <span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## check for nuclei with 0 marker expression</span></span>
<span><span class="va">zero_cell_filter</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MatrixGenerics/man/rowSums.html" class="external-link">colSums</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html" class="external-link">exprs</a></span><span class="op">(</span><span class="va">exp_set_sce</span><span class="op">)</span><span class="op">)</span> <span class="op">!=</span> <span class="fl">0</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">message</a></span><span class="op">(</span><span class="st">"Exclude "</span>, <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="op">!</span><span class="va">zero_cell_filter</span><span class="op">)</span>, <span class="st">" cells"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Exclude 0 cells</span></span>
<span></span>
<span><span class="va">exp_set_sce</span> <span class="op">&lt;-</span> <span class="va">exp_set_sce</span><span class="op">[</span>, <span class="va">zero_cell_filter</span><span class="op">]</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="run-bisque">Run Bisque<a class="anchor" aria-label="anchor" href="#run-bisque"></a>
</h3>
<p><code>Bisque</code> needs the bulk and single cell
<code>ExpressionSet</code> we prepared above, plus columns in the single
cell data that specify the cell type annotation to use
<code>cellType_broad_hc</code> and donor id (<code>BrNum</code> in this
data).</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Run Bisque with bulk and single cell ExpressionSet inputs</span></span>
<span><span class="va">est_prop</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BisqueRNA/man/ReferenceBasedDecomposition.html" class="external-link">ReferenceBasedDecomposition</a></span><span class="op">(</span></span>
<span>    bulk.eset <span class="op">=</span> <span class="va">exp_set_bulk</span>,</span>
<span>    sc.eset <span class="op">=</span> <span class="va">exp_set_sce</span>,</span>
<span>    cell.types <span class="op">=</span> <span class="st">"cellType_broad_hc"</span>,</span>
<span>    subject.names <span class="op">=</span> <span class="st">"BrNum"</span>,</span>
<span>    use.overlap <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Decomposing into 7 cell types.</span></span>
<span><span class="co">#&gt; Using 161 genes in both bulk and single-cell expression.</span></span>
<span><span class="co">#&gt; Converting single-cell counts to CPM and filtering zero variance genes.</span></span>
<span><span class="co">#&gt; Filtered 0 zero variance genes.</span></span>
<span><span class="co">#&gt; Converting bulk counts to CPM and filtering unexpressed genes.</span></span>
<span><span class="co">#&gt; Filtered 0 unexpressed genes.</span></span>
<span><span class="co">#&gt; Generating single-cell based reference from 56447 cells.</span></span>
<span><span class="co">#&gt; Inferring bulk transformation from single-cell alone.</span></span>
<span><span class="co">#&gt; Applying transformation to bulk samples and decomposing.</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="explore-output">Explore Output<a class="anchor" aria-label="anchor" href="#explore-output"></a>
</h3>
<p><code>Bisque</code> predicts the proportion of the cell types in
<code>cellType_broad_hc</code> for each sample in the bulk data.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Examine the output from Bisque, transpose to make it easier to work with</span></span>
<span><span class="va">est_prop</span><span class="op">$</span><span class="va">bulk.props</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">est_prop</span><span class="op">$</span><span class="va">bulk.props</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## sample x cell type matrix</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">est_prop</span><span class="op">$</span><span class="va">bulk.props</span><span class="op">)</span></span>
<span><span class="co">#&gt;                                    Astro  EndoMural      Micro     Oligo</span></span>
<span><span class="co">#&gt; 2107UNHS-0291_Br2720_Mid_Bulk 0.05786828 0.02373868 0.03522481 0.1885803</span></span>
<span><span class="co">#&gt; 2107UNHS-0291_Br2720_Mid_Cyto 0.09613503 0.01576541 0.04236801 0.2101065</span></span>
<span><span class="co">#&gt; 2107UNHS-0291_Br2720_Mid_Nuc  0.04995672 0.03686927 0.03709439 0.2216284</span></span>
<span><span class="co">#&gt; 2107UNHS-0291_Br6432_Ant_Bulk 0.06862682 0.03756306 0.03297615 0.1937775</span></span>
<span><span class="co">#&gt; 2107UNHS-0291_Br6432_Ant_Cyto 0.09860176 0.03144221 0.04428703 0.2241753</span></span>
<span><span class="co">#&gt; 2107UNHS-0291_Br6432_Ant_Nuc  0.05886274 0.04694488 0.03311716 0.2245202</span></span>
<span><span class="co">#&gt;                                      OPC     Excit     Inhib</span></span>
<span><span class="co">#&gt; 2107UNHS-0291_Br2720_Mid_Bulk 0.04396245 0.4976930 0.1529325</span></span>
<span><span class="co">#&gt; 2107UNHS-0291_Br2720_Mid_Cyto 0.01578266 0.4728291 0.1470133</span></span>
<span><span class="co">#&gt; 2107UNHS-0291_Br2720_Mid_Nuc  0.01988676 0.4494565 0.1851079</span></span>
<span><span class="co">#&gt; 2107UNHS-0291_Br6432_Ant_Bulk 0.04199513 0.4474728 0.1775886</span></span>
<span><span class="co">#&gt; 2107UNHS-0291_Br6432_Ant_Cyto 0.02038414 0.4445342 0.1365754</span></span>
<span><span class="co">#&gt; 2107UNHS-0291_Br6432_Ant_Nuc  0.02330603 0.4278448 0.1854042</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="explore-deconvolution-output-and-create-composition-plots-with-deconvobuddies-tools">5. Explore deconvolution output and create composition plots with
<code>DeconvoBuddies</code> tools<a class="anchor" aria-label="anchor" href="#explore-deconvolution-output-and-create-composition-plots-with-deconvobuddies-tools"></a>
</h2>
<p>To visualize the cell type proportion predictions, we can plot cell
type composition bar plots with
<code><a href="../reference/plot_composition_bar.html">DeconvoBuddies::plot_composition_bar()</a></code>, either the
prediction for each sample, or the average proportion over a group of
samples.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## add Phenotype data to proportion estimates</span></span>
<span><span class="va">pd</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">SAMPLE_ID</span>, <span class="va">Sample</span>, <span class="va">library_combo</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## make proportion estimates long so they are ggplot2 friendly</span></span>
<span><span class="va">prop_long</span> <span class="op">&lt;-</span> <span class="va">est_prop</span><span class="op">$</span><span class="va">bulk.props</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">rownames_to_column</a></span><span class="op">(</span><span class="st">"SAMPLE_ID"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">tidyr</span><span class="fu">::</span><span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_longer.html" class="external-link">pivot_longer</a></span><span class="op">(</span><span class="op">!</span><span class="va">SAMPLE_ID</span>, names_to <span class="op">=</span> <span class="st">"cell_type"</span>, values_to <span class="op">=</span> <span class="st">"prop"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span><span class="va">pd</span><span class="op">)</span></span>
<span><span class="co">#&gt; Joining with `by = join_by(SAMPLE_ID)`</span></span>
<span></span>
<span><span class="co">## create composition bar plots</span></span>
<span></span>
<span><span class="co">## for all library preparations by sample n=110</span></span>
<span><span class="co">## Remove the SAMPLE_ID names since they are very long using ggplot2::theme()</span></span>
<span><span class="fu"><a href="../reference/plot_composition_bar.html">plot_composition_bar</a></span><span class="op">(</span></span>
<span>    prop_long <span class="op">=</span> <span class="va">prop_long</span>,</span>
<span>    sample_col <span class="op">=</span> <span class="st">"SAMPLE_ID"</span>,</span>
<span>    x_col <span class="op">=</span> <span class="st">"SAMPLE_ID"</span>,</span>
<span>    add_text <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>, axis.ticks.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="Deconvolution_Benchmark_DLPFC_files/figure-html/composition%20plots-1.png" width="700"></p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co">## Average by brain donor</span></span>
<span><span class="fu"><a href="../reference/plot_composition_bar.html">plot_composition_bar</a></span><span class="op">(</span></span>
<span>    prop_long <span class="op">=</span> <span class="va">prop_long</span>,</span>
<span>    sample_col <span class="op">=</span> <span class="st">"SAMPLE_ID"</span>,</span>
<span>    x_col <span class="op">=</span> <span class="st">"Sample"</span>,</span>
<span>    add_text <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="Deconvolution_Benchmark_DLPFC_files/figure-html/composition%20plots-2.png" width="700"></p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co">## Each brain donor has up to 6 unique RNA library type and RNA extraction</span></span>
<span><span class="co">## combinations </span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">prop_long</span><span class="op">$</span><span class="va">Sample</span><span class="op">)</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">prop_long</span><span class="op">$</span><span class="va">cell_type</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  Br2720_mid Br2720_post  Br2743_ant  Br3942_ant  Br3942_mid  Br6423_ant </span></span>
<span><span class="co">#&gt;           5           6           5           6           6           6 </span></span>
<span><span class="co">#&gt; Br6423_post  Br6432_ant  Br6432_mid  Br6471_ant  Br6471_mid  Br6522_mid </span></span>
<span><span class="co">#&gt;           6           6           6           6           6           6 </span></span>
<span><span class="co">#&gt; Br6522_post  Br8325_ant  Br8325_mid  Br8492_mid Br8492_post  Br8667_ant </span></span>
<span><span class="co">#&gt;           6           6           5           5           6           6 </span></span>
<span><span class="co">#&gt;  Br8667_mid </span></span>
<span><span class="co">#&gt;           6</span></span>
<span></span>
<span><span class="co">## Here are the 6 "SAMPLE_ID" values for brain donor with ID "Br8667_mid"</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">prop_long</span><span class="op">$</span><span class="va">SAMPLE_ID</span><span class="op">[</span><span class="va">prop_long</span><span class="op">$</span><span class="va">Sample</span> <span class="op">==</span> <span class="st">"Br8667_mid"</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "AN00000904_Br8667_Mid_Bulk" "AN00000904_Br8667_Mid_Cyto"</span></span>
<span><span class="co">#&gt; [3] "AN00000904_Br8667_Mid_Nuc"  "AN00000906_Br8667_Mid_Bulk"</span></span>
<span><span class="co">#&gt; [5] "AN00000906_Br8667_Mid_Cyto" "AN00000906_Br8667_Mid_Nuc"</span></span></code></pre></div>
<p>This is a more complex scenario than the one from the <a href="https://research.libd.org/DeconvoBuddies/articles/DeconvoBuddies.html" class="external-link">introductory
vignette</a> where we were using random data. The first plot shows each
of the n=110 bulk RNA-seq samples we have. The second plot shows the
composition using the average across the 6 RNA library types and RNA
extractions for each brain donor.</p>
</div>
<div class="section level2">
<h2 id="check-proportion-against-rnascopeif-estimated-proportions">6. Check proportion against RNAScope/IF estimated proportions<a class="anchor" aria-label="anchor" href="#check-proportion-against-rnascopeif-estimated-proportions"></a>
</h2>
<p>Note to compare the deconvolution results to the RNAScope/IF
proportions, Oligo and OPC need to be added together.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Combine Oligo and OPC into OligoOPC</span></span>
<span><span class="va">prop_long_opc</span> <span class="op">&lt;-</span> <span class="va">prop_long</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>cell_type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"Oligo|OPC"</span>, <span class="st">"OligoOPC"</span>, <span class="va">cell_type</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">SAMPLE_ID</span>, <span class="va">Sample</span>, <span class="va">library_combo</span>, <span class="va">cell_type</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarize</a></span><span class="op">(</span>prop <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">prop</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">ungroup</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; `summarise()` has grouped output by 'SAMPLE_ID', 'Sample', 'library_combo'. You</span></span>
<span><span class="co">#&gt; can override using the `.groups` argument.</span></span>
<span></span>
<span><span class="va">prop_long_opc</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html" class="external-link">count</a></span><span class="op">(</span><span class="va">cell_type</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 2</span></span></span>
<span><span class="co">#&gt;   cell_type     n</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> Astro       110</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> EndoMural   110</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> Excit       110</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> Inhib       110</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> Micro       110</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> OligoOPC    110</span></span>
<span></span>
<span><span class="co">## Join RNAScope/IF and Bisque cell type proportions</span></span>
<span><span class="va">prop_compare</span> <span class="op">&lt;-</span> <span class="va">prop_long_opc</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">inner_join</a></span><span class="op">(</span></span>
<span>        <span class="va">RNAScope_prop</span> <span class="op">|&gt;</span></span>
<span>            <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">Sample</span>, <span class="va">cell_type</span>, prop_RNAScope <span class="op">=</span> <span class="va">prop</span>, <span class="va">prop_sn</span><span class="op">)</span>,</span>
<span>        by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Sample"</span>, <span class="st">"cell_type"</span><span class="op">)</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<p>We can now calculate the correlation plot a scatter plot of the
proportions. Note that you can change the type of correlation algorithm
used in the <code><a href="https://rdrr.io/r/stats/cor.html" class="external-link">cor()</a></code> function. The default method is
<code>"pearson"</code>.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## compute correlation with RNAScope/IF proportions</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/stats/cor.html" class="external-link">cor</a></span><span class="op">(</span><span class="va">prop_compare</span><span class="op">$</span><span class="va">prop</span>, <span class="va">prop_compare</span><span class="op">$</span><span class="va">prop_RNAScope</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 0.5357834</span></span>
<span></span>
<span><span class="co">## Scatter plot with RNAScope/IF proportions</span></span>
<span><span class="va">prop_compare</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">prop_RNAScope</span>, y <span class="op">=</span> <span class="va">prop</span>, color <span class="op">=</span> <span class="va">cell_type</span>, shape <span class="op">=</span> <span class="va">library_combo</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_abline</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="Deconvolution_Benchmark_DLPFC_files/figure-html/proportion%20scatter-1.png" width="700"></p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span></span>
<span><span class="co">## correlation with snRNA-seq proportion</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/stats/cor.html" class="external-link">cor</a></span><span class="op">(</span><span class="va">prop_compare</span><span class="op">$</span><span class="va">prop</span>, <span class="va">prop_compare</span><span class="op">$</span><span class="va">prop_sn</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 0.8397746</span></span>
<span></span>
<span><span class="co">## Scatter plot with RNAScope/IF proportions</span></span>
<span><span class="va">prop_compare</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">prop_sn</span>, y <span class="op">=</span> <span class="va">prop</span>, color <span class="op">=</span> <span class="va">cell_type</span>, shape <span class="op">=</span> <span class="va">library_combo</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_abline</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="Deconvolution_Benchmark_DLPFC_files/figure-html/proportion%20scatter-2.png" width="700"></p>
<p>In the first plot we can see how <code>Bisque</code> overestimates
the proportion of excitatory neurons (<code>Excit</code>) when compared
against RNAScope/IF as all <code>Excit</code> points are higher than the
diagonal line shown in black.</p>
<p>In the second plot we can see how <code>Bisque</code> proportions are
closer to the snRNA-seq proportions. After all, given the <a href="https://doi.org/10.1186/s13059-023-03123-4" class="external-link">challenges in
generating orthogonal cell type proportion data</a>, <code>Bisque</code>
and other deconvolution methods were developed by comparing against
simulated data, sometimes using pseudo-bulk sc/snRNA-seq data.</p>
</div>
<div class="section level2">
<h2 id="how-to-run-deconvolution-with-hspe">7. How to run deconvolution with <code>hspe</code><a class="anchor" aria-label="anchor" href="#how-to-run-deconvolution-with-hspe"></a>
</h2>
<p><em>hspe</em> (formerly called <em>dtangle</em>) is another R package
for deconvolution. It was also a top performing method in our <a href="https://github.com/LieberInstitute/Human_DLPFC_Deconvolution" class="external-link">deconvolution
benchmark</a>.</p>
<p><code>hspe</code> is <a href="https://gjhunt.github.io/hspe/" class="external-link">downloadable</a> from GitHub but
can’t be shown on this vignette as Bioconductor packages cannot use
packages from GitHub.</p>
<p>Below we show some example code to prepare input data and run
<code>hspe</code> (not run here):</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="co">## Install hspe</span></span>
<span>    <span class="co"># if (!requireNamespace("hspe", quietly = TRUE)) {</span></span>
<span>    <span class="co">#     ## Install version 0.1 which is the one listed on the main documentation</span></span>
<span>    <span class="co">#     ## at https://github.com/gjhunt/hspe/tree/main?tab=readme-ov-file#software</span></span>
<span>    <span class="co">#     remotes::install_url("https://github.com/gjhunt/hspe/raw/main/hspe_0.1.tar.gz")</span></span>
<span>    <span class="co">#     ## Alternatively, install from the latest version on GitHub with:</span></span>
<span>    <span class="co">#     # remotes::install_github("gjhunt/hspe", subdir = "lib_hspe")</span></span>
<span>    <span class="co">#     </span></span>
<span>    <span class="co">#     ## As of 2024-08-23, it's been 3 years since files were last modified</span></span>
<span>    <span class="co">#     ## at https://github.com/gjhunt/hspe/tree/main/lib_hspe. </span></span>
<span>    <span class="co"># }</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"hspe"</span><span class="op">)</span></span>
<span>    </span>
<span>    <span class="co"># pseudobulk the sce data by sample + cell type</span></span>
<span>    <span class="va">sce_pb</span> <span class="op">&lt;-</span> <span class="fu">spatialLIBD</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/spatialLIBD/man/registration_pseudobulk.html" class="external-link">registration_pseudobulk</a></span><span class="op">(</span><span class="va">sce</span>,</span>
<span>        var_registration <span class="op">=</span> <span class="st">"cellType_broad_hc"</span>,</span>
<span>        var_sample_id <span class="op">=</span> <span class="st">"Sample"</span></span>
<span>    <span class="op">)</span></span>
<span></span>
<span>    <span class="co">## extract the gene expression from the bulk rse_gene</span></span>
<span>    <span class="va">mixture_samples</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/t.html" class="external-link">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span><span class="op">$</span><span class="va">logcounts</span><span class="op">)</span></span>
<span>    <span class="va">mixture_samples</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span>
<span></span>
<span>    <span class="co">## create a vector of indexes of the different cell types</span></span>
<span>    <span class="va">pure_samples</span> <span class="op">&lt;-</span> <span class="fu">rafalib</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/rafalib/man/splitit.html" class="external-link">splitit</a></span><span class="op">(</span><span class="va">sce_pb</span><span class="op">$</span><span class="va">cellType_broad_hc</span><span class="op">)</span></span>
<span></span>
<span>    <span class="co">## extract the the pseudobulked logcounts</span></span>
<span>    <span class="va">reference_samples</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/t.html" class="external-link">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">sce_pb</span><span class="op">)</span><span class="op">$</span><span class="va">logcounts</span><span class="op">)</span></span>
<span>    <span class="va">reference_samples</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span>
<span></span>
<span>    <span class="co">## check the number of genes match in the bulk (mixture) and single cell (reference)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">mixture_samples</span><span class="op">)</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">reference_samples</span><span class="op">)</span></span>
<span></span>
<span>    <span class="co">## run hspe</span></span>
<span>    <span class="va">est_prop_hspe</span> <span class="op">&lt;-</span> <span class="fu">hspe</span><span class="op">(</span></span>
<span>        Y <span class="op">=</span> <span class="va">mixture_samples</span>,</span>
<span>        reference <span class="op">=</span> <span class="va">reference_samples</span>,</span>
<span>        pure_samples <span class="op">=</span> <span class="va">pure_samples</span>,</span>
<span>        markers <span class="op">=</span> <span class="va">marker_genes</span>,</span>
<span>        seed <span class="op">=</span> <span class="fl">10524</span></span>
<span>    <span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="conclusion">Conclusion<a class="anchor" aria-label="anchor" href="#conclusion"></a>
</h2>
<p>In this vignette we have demonstrated some of the functions and data
in the <code>DeconvoBuddies</code> package, and how to use them in a
deconvolution workflow to predict cell type proportions. We used real
data from a study on <a href="https://github.com/LieberInstitute/Human_DLPFC_Deconvolution" class="external-link">human
brain</a>.</p>
</div>
<div class="section level2">
<h2 id="reproducibility">Reproducibility<a class="anchor" aria-label="anchor" href="#reproducibility"></a>
</h2>
<p>The <em><a href="https://bioconductor.org/packages/3.20/DeconvoBuddies" class="external-link">DeconvoBuddies</a></em>
package (Huuki-Myers, Maynard, Hicks, Zandi, Kleinman, Hyde, Goes, and
Collado-Torres, 2024) was made possible thanks to:</p>
<ul>
<li>R (R Core Team, 2024)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.20/BiocStyle" class="external-link">BiocStyle</a></em>
(Oleś, 2024)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=knitr" class="external-link">knitr</a></em> (Xie,
2024)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=RefManageR" class="external-link">RefManageR</a></em>
(McLean, 2017)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=rmarkdown" class="external-link">rmarkdown</a></em>
(Allaire, Xie, Dervieux, McPherson, Luraschi, Ushey, Atkins, Wickham,
Cheng, Chang, and Iannone, 2024)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=sessioninfo" class="external-link">sessioninfo</a></em>
(Wickham, Chang, Flight, Müller, and Hester, 2021)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=testthat" class="external-link">testthat</a></em>
(Wickham, 2011)</li>
</ul>
<p>This package was developed using <em><a href="https://bioconductor.org/packages/3.20/biocthis" class="external-link">biocthis</a></em>.</p>
<p>Code for creating the vignette</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Create the vignette</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/rstudio/rmarkdown" class="external-link">"rmarkdown"</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/system.time.html" class="external-link">system.time</a></span><span class="op">(</span><span class="fu"><a href="https://pkgs.rstudio.com/rmarkdown/reference/render.html" class="external-link">render</a></span><span class="op">(</span><span class="st">"Deconvolution_Benchmark_DLPFC.Rmd"</span>, <span class="st">"BiocStyle::html_document"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Extract the R code</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://yihui.org/knitr/" class="external-link">"knitr"</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/knit.html" class="external-link">knit</a></span><span class="op">(</span><span class="st">"Deconvolution_Benchmark_DLPFC.Rmd"</span>, tangle <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>Date the vignette was generated.</p>
<pre><code><span><span class="co">#&gt; [1] "2024-08-23 22:33:29 UTC"</span></span></code></pre>
<p>Wallclock time spent generating the vignette.</p>
<pre><code><span><span class="co">#&gt; Time difference of 1.259 mins</span></span></code></pre>
<p><code>R</code> session information.</p>
<pre><code><span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt;  <span style="color: #555555; font-style: italic;">setting </span> <span style="color: #555555; font-style: italic;">value</span></span></span>
<span><span class="co">#&gt;  version  R version 4.4.1 (2024-06-14)</span></span>
<span><span class="co">#&gt;  os       Ubuntu 22.04.4 LTS</span></span>
<span><span class="co">#&gt;  system   x86_64, linux-gnu</span></span>
<span><span class="co">#&gt;  ui       X11</span></span>
<span><span class="co">#&gt;  language en</span></span>
<span><span class="co">#&gt;  collate  en_US.UTF-8</span></span>
<span><span class="co">#&gt;  ctype    en_US.UTF-8</span></span>
<span><span class="co">#&gt;  tz       UTC</span></span>
<span><span class="co">#&gt;  date     2024-08-23</span></span>
<span><span class="co">#&gt;  pandoc   3.3 @ /usr/bin/ (via rmarkdown)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt;  <span style="color: #555555; font-style: italic;">package             </span> <span style="color: #555555; font-style: italic;">*</span> <span style="color: #555555; font-style: italic;">version   </span> <span style="color: #555555; font-style: italic;">date (UTC)</span> <span style="color: #555555; font-style: italic;">lib</span> <span style="color: #555555; font-style: italic;">source</span></span></span>
<span><span class="co">#&gt;  abind                  1.4-5      <span style="color: #555555;">2016-07-21</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  AnnotationDbi          1.67.0     <span style="color: #555555;">2024-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  AnnotationHub          3.13.3     <span style="color: #555555;">2024-08-19</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  attempt                0.3.1      <span style="color: #555555;">2020-05-03</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  backports              1.5.0      <span style="color: #555555;">2024-05-23</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  beachmat               2.21.5     <span style="color: #555555;">2024-07-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  beeswarm               0.4.0      <span style="color: #555555;">2021-06-01</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  benchmarkme            1.0.8      <span style="color: #555555;">2022-06-12</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  benchmarkmeData        1.0.4      <span style="color: #555555;">2020-04-23</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  bibtex                 0.5.1      <span style="color: #555555;">2023-01-26</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  Biobase              * 2.65.0     <span style="color: #555555;">2024-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  BiocFileCache          2.13.0     <span style="color: #555555;">2024-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  BiocGenerics         * 0.51.0     <span style="color: #555555;">2024-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  BiocIO                 1.15.1     <span style="color: #555555;">2024-08-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  BiocManager            1.30.24    <span style="color: #555555;">2024-08-20</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  BiocNeighbors          1.23.0     <span style="color: #555555;">2024-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  BiocParallel           1.39.0     <span style="color: #555555;">2024-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  BiocSingular           1.21.2     <span style="color: #555555;">2024-07-04</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  BiocStyle            * 2.33.1     <span style="color: #555555;">2024-06-12</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  BiocVersion            3.20.0     <span style="color: #555555;">2024-05-01</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  Biostrings             2.73.1     <span style="color: #555555;">2024-06-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  BisqueRNA            * 1.0.5      <span style="color: #555555;">2021-05-23</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  bit                    4.0.5      <span style="color: #555555;">2022-11-15</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  bit64                  4.0.5      <span style="color: #555555;">2020-08-30</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  bitops                 1.0-8      <span style="color: #555555;">2024-07-29</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  blob                   1.2.4      <span style="color: #555555;">2023-03-17</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  bluster                1.15.0     <span style="color: #555555;">2024-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  bookdown               0.40       <span style="color: #555555;">2024-07-02</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  bslib                  0.8.0      <span style="color: #555555;">2024-07-29</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  cachem                 1.1.0      <span style="color: #555555;">2024-05-16</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  cli                    3.6.3      <span style="color: #555555;">2024-06-21</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  cluster                2.1.6      <span style="color: #555555;">2023-12-01</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  codetools              0.2-20     <span style="color: #555555;">2024-03-31</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  colorspace             2.1-1      <span style="color: #555555;">2024-07-26</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  config                 0.3.2      <span style="color: #555555;">2023-08-30</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  cowplot                1.1.3      <span style="color: #555555;">2024-01-22</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  crayon                 1.5.3      <span style="color: #555555;">2024-06-20</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  curl                   5.2.1      <span style="color: #555555;">2024-03-01</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  data.table             1.15.4     <span style="color: #555555;">2024-03-30</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  DBI                    1.2.3      <span style="color: #555555;">2024-06-02</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  dbplyr                 2.5.0      <span style="color: #555555;">2024-03-19</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  DeconvoBuddies       * 0.99.0     <span style="color: #555555;">2024-08-23</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  DelayedArray           0.31.11    <span style="color: #555555;">2024-08-04</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  desc                   1.4.3      <span style="color: #555555;">2023-12-10</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  digest                 0.6.37     <span style="color: #555555;">2024-08-19</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  doParallel             1.0.17     <span style="color: #555555;">2022-02-07</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  dotCall64              1.1-1      <span style="color: #555555;">2023-11-28</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  dplyr                * 1.1.4      <span style="color: #555555;">2023-11-17</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  dqrng                  0.4.1      <span style="color: #555555;">2024-05-28</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  DT                     0.33       <span style="color: #555555;">2024-04-04</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  edgeR                  4.3.11     <span style="color: #555555;">2024-08-19</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  evaluate               0.24.0     <span style="color: #555555;">2024-06-10</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  ExperimentHub          2.13.1     <span style="color: #555555;">2024-07-31</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  fansi                  1.0.6      <span style="color: #555555;">2023-12-08</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  farver                 2.1.2      <span style="color: #555555;">2024-05-13</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  fastmap                1.2.0      <span style="color: #555555;">2024-05-15</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  fields                 16.2       <span style="color: #555555;">2024-06-27</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  filelock               1.0.3      <span style="color: #555555;">2023-12-11</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  foreach                1.5.2      <span style="color: #555555;">2022-02-02</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  fs                     1.6.4      <span style="color: #555555;">2024-04-25</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  generics               0.1.3      <span style="color: #555555;">2022-07-05</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  GenomeInfoDb         * 1.41.1     <span style="color: #555555;">2024-05-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  GenomeInfoDbData       1.2.12     <span style="color: #555555;">2024-06-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  GenomicAlignments      1.41.0     <span style="color: #555555;">2024-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  GenomicRanges        * 1.57.1     <span style="color: #555555;">2024-06-12</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  ggbeeswarm             0.7.2      <span style="color: #555555;">2023-04-29</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  ggplot2              * 3.5.1      <span style="color: #555555;">2024-04-23</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  ggrepel                0.9.5      <span style="color: #555555;">2024-01-10</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  glue                   1.7.0      <span style="color: #555555;">2024-01-09</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  golem                  0.5.0      <span style="color: #555555;">2024-08-19</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  gridExtra              2.3        <span style="color: #555555;">2017-09-09</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  gtable                 0.3.5      <span style="color: #555555;">2024-04-22</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  HDF5Array              1.33.6     <span style="color: #555555;">2024-08-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  highr                  0.11       <span style="color: #555555;">2024-05-26</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  htmltools              0.5.8.1    <span style="color: #555555;">2024-04-04</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  htmlwidgets            1.6.4      <span style="color: #555555;">2023-12-06</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  httpuv                 1.6.15     <span style="color: #555555;">2024-03-26</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  httr                   1.4.7      <span style="color: #555555;">2023-08-15</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  igraph                 2.0.3      <span style="color: #555555;">2024-03-13</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  IRanges              * 2.39.2     <span style="color: #555555;">2024-07-17</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  irlba                  2.3.5.1    <span style="color: #555555;">2022-10-03</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  iterators              1.0.14     <span style="color: #555555;">2022-02-05</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  jquerylib              0.1.4      <span style="color: #555555;">2021-04-26</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  jsonlite               1.8.8      <span style="color: #555555;">2023-12-04</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  KEGGREST               1.45.1     <span style="color: #555555;">2024-06-17</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  knitr                  1.48       <span style="color: #555555;">2024-07-07</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  labeling               0.4.3      <span style="color: #555555;">2023-08-29</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  later                  1.3.2      <span style="color: #555555;">2023-12-06</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  lattice                0.22-6     <span style="color: #555555;">2024-03-20</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  lazyeval               0.2.2      <span style="color: #555555;">2019-03-15</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  lifecycle              1.0.4      <span style="color: #555555;">2023-11-07</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  limma                  3.61.9     <span style="color: #555555;">2024-08-04</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  limSolve               1.5.7.1    <span style="color: #555555;">2024-02-05</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  locfit                 1.5-9.10   <span style="color: #555555;">2024-06-24</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  lpSolve                5.6.20     <span style="color: #555555;">2023-12-10</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  lubridate              1.9.3      <span style="color: #555555;">2023-09-27</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  magick                 2.8.4      <span style="color: #555555;">2024-07-14</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  magrittr               2.0.3      <span style="color: #555555;">2022-03-30</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  maps                   3.4.2      <span style="color: #555555;">2023-12-15</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  MASS                   7.3-61     <span style="color: #555555;">2024-06-13</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  Matrix                 1.7-0      <span style="color: #555555;">2024-04-26</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  MatrixGenerics       * 1.17.0     <span style="color: #555555;">2024-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  matrixStats          * 1.3.0      <span style="color: #555555;">2024-04-11</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  memoise                2.0.1      <span style="color: #555555;">2021-11-26</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  metapod                1.13.0     <span style="color: #555555;">2024-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  mime                   0.12       <span style="color: #555555;">2021-09-28</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  munsell                0.5.1      <span style="color: #555555;">2024-04-01</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  paletteer              1.6.0      <span style="color: #555555;">2024-01-21</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  pillar                 1.9.0      <span style="color: #555555;">2023-03-22</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  pkgconfig              2.0.3      <span style="color: #555555;">2019-09-22</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  pkgdown                <span style="color: #BB00BB; font-weight: bold;">2.1.0.9000</span> <span style="color: #555555;">2024-08-23</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">Github (r-lib/pkgdown@020dc44)</span></span></span>
<span><span class="co">#&gt;  plotly                 4.10.4     <span style="color: #555555;">2024-01-13</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  plyr                   1.8.9      <span style="color: #555555;">2023-10-02</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  png                    0.1-8      <span style="color: #555555;">2022-11-29</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  promises               1.3.0      <span style="color: #555555;">2024-04-05</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  purrr                  1.0.2      <span style="color: #555555;">2023-08-10</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  quadprog               1.5-8      <span style="color: #555555;">2019-11-20</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  R6                     2.5.1      <span style="color: #555555;">2021-08-19</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  rafalib                1.0.0      <span style="color: #555555;">2015-08-09</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  ragg                   1.3.2      <span style="color: #555555;">2024-05-15</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  rappdirs               0.3.3      <span style="color: #555555;">2021-01-31</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  RColorBrewer           1.1-3      <span style="color: #555555;">2022-04-03</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  Rcpp                   1.0.13     <span style="color: #555555;">2024-07-17</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  RCurl                  1.98-1.16  <span style="color: #555555;">2024-07-11</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  RefManageR           * 1.4.0      <span style="color: #555555;">2022-09-30</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  rematch2               2.1.2      <span style="color: #555555;">2020-05-01</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  reshape2               1.4.4      <span style="color: #555555;">2020-04-09</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  restfulr               0.0.15     <span style="color: #555555;">2022-06-16</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  rhdf5                  2.49.0     <span style="color: #555555;">2024-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  rhdf5filters           1.17.0     <span style="color: #555555;">2024-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  Rhdf5lib               1.27.0     <span style="color: #555555;">2024-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  rjson                  0.2.22     <span style="color: #555555;">2024-08-20</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  rlang                  1.1.4      <span style="color: #555555;">2024-06-04</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  rmarkdown              2.28       <span style="color: #555555;">2024-08-17</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  Rsamtools              2.21.1     <span style="color: #555555;">2024-08-16</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  RSQLite                2.3.7      <span style="color: #555555;">2024-05-27</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  rsvd                   1.0.5      <span style="color: #555555;">2021-04-16</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  rtracklayer            1.65.0     <span style="color: #555555;">2024-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  S4Arrays               1.5.7      <span style="color: #555555;">2024-08-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  S4Vectors            * 0.43.2     <span style="color: #555555;">2024-07-17</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  sass                   0.4.9      <span style="color: #555555;">2024-03-15</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  ScaledMatrix           1.13.0     <span style="color: #555555;">2024-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  scales                 1.3.0      <span style="color: #555555;">2023-11-28</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  scater                 1.33.4     <span style="color: #555555;">2024-07-21</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  scran                  1.33.1     <span style="color: #555555;">2024-07-23</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  scuttle                1.15.4     <span style="color: #555555;">2024-08-14</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  sessioninfo          * 1.2.2      <span style="color: #555555;">2021-12-06</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  shiny                  1.9.1      <span style="color: #555555;">2024-08-01</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  shinyWidgets           0.8.6      <span style="color: #555555;">2024-04-24</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  SingleCellExperiment * 1.27.2     <span style="color: #555555;">2024-05-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  spam                   2.10-0     <span style="color: #555555;">2023-10-23</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  SparseArray            1.5.31     <span style="color: #555555;">2024-08-04</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  SpatialExperiment    * 1.15.1     <span style="color: #555555;">2024-06-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  spatialLIBD          * 1.17.8     <span style="color: #555555;">2024-07-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  statmod                1.5.0      <span style="color: #555555;">2023-01-06</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  stringi                1.8.4      <span style="color: #555555;">2024-05-06</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  stringr                1.5.1      <span style="color: #555555;">2023-11-14</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  SummarizedExperiment * 1.35.1     <span style="color: #555555;">2024-06-28</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  systemfonts            1.1.0      <span style="color: #555555;">2024-05-15</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  textshaping            0.4.0      <span style="color: #555555;">2024-05-24</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  tibble               * 3.2.1      <span style="color: #555555;">2023-03-20</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  tidyr                * 1.3.1      <span style="color: #555555;">2024-01-24</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  tidyselect             1.2.1      <span style="color: #555555;">2024-03-11</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  timechange             0.3.0      <span style="color: #555555;">2024-01-18</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  UCSC.utils             1.1.0      <span style="color: #555555;">2024-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  utf8                   1.2.4      <span style="color: #555555;">2023-10-22</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  vctrs                  0.6.5      <span style="color: #555555;">2023-12-01</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  vipor                  0.4.7      <span style="color: #555555;">2023-12-18</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  viridis                0.6.5      <span style="color: #555555;">2024-01-29</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  viridisLite            0.4.2      <span style="color: #555555;">2023-05-02</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  withr                  3.0.1      <span style="color: #555555;">2024-07-31</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  xfun                   0.47       <span style="color: #555555;">2024-08-17</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  XML                    3.99-0.17  <span style="color: #555555;">2024-06-25</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  xml2                   1.3.6      <span style="color: #555555;">2023-12-04</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  xtable                 1.8-4      <span style="color: #555555;">2019-04-21</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  XVector                0.45.0     <span style="color: #555555;">2024-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  yaml                   2.3.10     <span style="color: #555555;">2024-07-26</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  zlibbioc               1.51.1     <span style="color: #555555;">2024-06-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [1] /__w/_temp/Library</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [2] /usr/local/lib/R/site-library</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [3] /usr/local/lib/R/library</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></span></code></pre>
</div>
<div class="section level2">
<h2 id="bibliography">Bibliography<a class="anchor" aria-label="anchor" href="#bibliography"></a>
</h2>
<p>This vignette was generated using <em><a href="https://bioconductor.org/packages/3.20/BiocStyle" class="external-link">BiocStyle</a></em>
(Oleś, 2024) with <em><a href="https://CRAN.R-project.org/package=knitr" class="external-link">knitr</a></em> (Xie,
2024) and <em><a href="https://CRAN.R-project.org/package=rmarkdown" class="external-link">rmarkdown</a></em>
(Allaire, Xie, Dervieux et al., 2024) running behind the scenes.</p>
<p>Citations made with <em><a href="https://CRAN.R-project.org/package=RefManageR" class="external-link">RefManageR</a></em>
(McLean, 2017).</p>
<p>
<a id="bib-allaire2024rmarkdown"></a><a href="#cite-allaire2024rmarkdown">[1]</a><cite>
J. Allaire, Y. Xie, C. Dervieux, et al. <em>rmarkdown: Dynamic Documents
for R</em>. R package version 2.28. 2024. URL:
<a href="https://github.com/rstudio/rmarkdown" class="external-link">https://github.com/rstudio/rmarkdown</a>.</cite>
</p>
<p>
<a id="bib-huukimyers2024deconvobuddies"></a><a href="#cite-huukimyers2024deconvobuddies">[2]</a><cite>
L. A. Huuki-Myers, K. R. Maynard, S. C. Hicks, et al.
<em>DeconvoBuddies: a R/Bioconductor package with deconvolution helper
functions</em>. <a href="https://github.com/LieberInstitute/DeconvoBuddies/DeconvoBuddies" class="external-link uri">https://github.com/LieberInstitute/DeconvoBuddies/DeconvoBuddies</a>
- R package version 0.99.0. 2024. DOI:
<a href="https://doi.org/10.18129/B9.bioc.DeconvoBuddies" class="external-link">10.18129/B9.bioc.DeconvoBuddies</a>.
URL:
<a href="http://www.bioconductor.org/packages/DeconvoBuddies" class="external-link">http://www.bioconductor.org/packages/DeconvoBuddies</a>.</cite>
</p>
<p>
<a id="bib-mclean2017refmanager"></a><a href="#cite-mclean2017refmanager">[3]</a><cite>
M. W. McLean. “RefManageR: Import and Manage BibTeX and BibLaTeX
References in R”. In: <em>The Journal of Open Source Software</em>
(2017). DOI:
<a href="https://doi.org/10.21105/joss.00338" class="external-link">10.21105/joss.00338</a>.</cite>
</p>
<p>
<a id="bib-ole2024biocstyle"></a><a href="#cite-ole2024biocstyle">[4]</a><cite>
A. Oleś. <em>BiocStyle: Standard styles for vignettes and other
Bioconductor documents</em>. R package version 2.33.1. 2024. DOI:
<a href="https://doi.org/10.18129/B9.bioc.BiocStyle" class="external-link">10.18129/B9.bioc.BiocStyle</a>.
URL:
<a href="https://bioconductor.org/packages/BiocStyle" class="external-link">https://bioconductor.org/packages/BiocStyle</a>.</cite>
</p>
<p>
<a id="bib-2024language"></a><a href="#cite-2024language">[5]</a><cite>
R Core Team. <em>R: A Language and Environment for Statistical
Computing</em>. R Foundation for Statistical Computing. Vienna, Austria,
2024. URL:
<a href="https://www.R-project.org/" class="external-link">https://www.R-project.org/</a>.</cite>
</p>
<p>
<a id="bib-wickham2011testthat"></a><a href="#cite-wickham2011testthat">[6]</a><cite>
H. Wickham. “testthat: Get Started with Testing”. In: <em>The R
Journal</em> 3 (2011), pp. 5–10. URL:
<a href="https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf" class="external-link">https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf</a>.</cite>
</p>
<p>
<a id="bib-wickham2021sessioninfo"></a><a href="#cite-wickham2021sessioninfo">[7]</a><cite>
H. Wickham, W. Chang, R. Flight, et al. <em>sessioninfo: R Session
Information</em>. R package version 1.2.2, <a href="https://r-lib.github.io/sessioninfo/" class="external-link uri">https://r-lib.github.io/sessioninfo/</a>. 2021. URL:
<a href="https://github.com/r-lib/sessioninfo#readme" class="external-link">https://github.com/r-lib/sessioninfo#readme</a>.</cite>
</p>
<p>
<a id="bib-xie2024knitr"></a><a href="#cite-xie2024knitr">[8]</a><cite>
Y. Xie. <em>knitr: A General-Purpose Package for Dynamic Report
Generation in R</em>. R package version 1.48. 2024. URL:
<a href="https://yihui.org/knitr/" class="external-link">https://yihui.org/knitr/</a>.</cite>
</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Louise Huuki-Myers.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.9000.</p>
</div>

      </footer>
</div>






  </body>
</html>
