<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>2. Finding Marker Genes with DeconvoBuddies • DeconvoBuddies</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="2. Finding Marker Genes with DeconvoBuddies">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">DeconvoBuddies</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/DeconvoBuddies.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Deconvolution_Benchmark_DLPFC.html">3. Deconvolution Benchmark in Human DLPFC</a>
    </li>
    <li>
      <a href="../articles/Marker_Finding.html">2. Finding Marker Genes with DeconvoBuddies</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/lahuuki/DeconvoBuddies/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>2. Finding Marker Genes with DeconvoBuddies</h1>
                        <h4 data-toc-skip class="author">Louise
Huuki-Myers</h4>
            <address class="author_afil">
      Lieber Institute for Brain Development, Johns Hopkins Medical
Campus<br><a class="author_email" href="mailto:#"></a><a href="mailto:lahuuki@gmail.com" class="email">lahuuki@gmail.com</a>
      </address>
                  
            <h4 data-toc-skip class="date">18 October 2024</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/lahuuki/DeconvoBuddies/blob/devel/vignettes/Marker_Finding.Rmd" class="external-link"><code>vignettes/Marker_Finding.Rmd</code></a></small>
      <div class="hidden name"><code>Marker_Finding.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<div class="section level3">
<h3 id="what-are-marker-genes">What are Marker Genes?<a class="anchor" aria-label="anchor" href="#what-are-marker-genes"></a>
</h3>
<p>Cell type marker genes have cell type specific expression, that is
high expression in the target cell type, and low expression in all other
cell types. Sub-setting the genes considered in a cell type
deconvolution analysis helps reduce noise and can improve the accuracy
of a deconvolution method.</p>
<div class="section level4">
<h4 id="how-can-we-select-marker-genes">How can we select marker genes?<a class="anchor" aria-label="anchor" href="#how-can-we-select-marker-genes"></a>
</h4>
<p>There are several approaches to select marker genes.</p>
<p>One popular method is “1 vs. All” differential expression <span class="citation">(Lun, McCarthy, and Marioni, n.d.)</span>, where genes
are tested for differential expression between the target cell type, and
a combined group of all <strong>“other”</strong> cell types.
Statistically significant differentially expressed genes (DEGs) can be
selected as a set of marker genes, DEGs can be ranked by high log fold
change.</p>
<p>However in some cases <em>1vAll</em> can select genes with high
expression in non-target cell types, especially in cell types related to
the target cell types (such as Neuron sub-types), or when there is a
smaller number of cells in the cell type and the signal is disguised
within the other group.</p>
<p>For example, in our <a href="https://github.com/LieberInstitute/Human_DLPFC_Deconvolution" class="external-link">snRNA-seq
dataset from Human DLPFC</a> <span class="citation">(Huuki-Myers et al.
2024)</span> selecting marker gene for the cell type Oligodendrocyte
(Oligo), <em>MBP</em> has a high log fold change when testing by
<em>1vALL</em> (see illustration below). But, when the expression of MBP
is observed by individual cell types there is also expression in the
related cell types Microglia (Micro) and Oligodendrocyte precursor cells
(OPC).</p>
<p align="center">
</p>
<div class="float">
<img src="http://research.libd.org/DeconvoBuddies/reference/figures/get_mean_ratio.png" alt="Figure 1. 1vALL Marker Gene differential expression vs. MeanRatio method for marker gene selection."><div class="figcaption">Figure 1. <em>1vALL</em> Marker Gene
differential expression vs. <em>MeanRatio</em> method for marker gene
selection.</div>
</div>

</div>
</div>
<div class="section level3">
<h3 id="the-mean-ratio-method">The <em>Mean Ratio</em> Method<a class="anchor" aria-label="anchor" href="#the-mean-ratio-method"></a>
</h3>
<p>To capture genes with more cell type specific expression and less
noise, we developed the <strong><em>Mean Ratio</em></strong> method. The
<em>Mean Ratio</em> method works by selecting genes with large
differences between gene expression in the target cell type and the
closest non-target cell type, by evaluating genes by their
<code>MeanRatio</code> metric.</p>
<p>We calculate the <code>MeanRatio</code> for a target cell type for
each gene by <strong>dividing the mean expression of the target cell by
the mean expression of the next highest non-target cell type</strong>.
Genes with the highest <code>MeanRatio</code> values are selected as
marker genes.</p>
<p>In the above example, <strong>Oligo</strong> is the target cell type.
Micro has the highest mean expression out of the other non-target (not
Oligo) cell types. The
<code>MeanRatio = (mean expression Oligo) / (mean expression Micro)</code>,
for <em>MBP</em> <code>MeanRatio</code> = 2.68 for gene <em>FOLH1</em>
<code>MeanRatio</code> is much higher 21.6 showing FOLH1 is the better
marker gene (in contrast to ranking by <em>1vALL</em> log FC). In the
violin plots you can see that expression of <em>FOLH1</em> is much more
specific to Oligo than MBP, supporting the ranking by
<code>MeanRatio</code>.</p>
<p>We have implemented the <strong><em>Mean Ratio</em></strong> method
in this R package with the function <code><a href="../reference/get_mean_ratio.html">get_mean_ratio()</a></code>. This
vignette will cover our process for marker gene selection.</p>
</div>
<div class="section level3">
<h3 id="goals-of-this-vignette">Goals of this Vignette<a class="anchor" aria-label="anchor" href="#goals-of-this-vignette"></a>
</h3>
<p>We will be demonstrating how to use <code>DeconvoBuddies</code> tools
when finding cell type marker genes in single cell RNA-seq data via the
<em>MeanRatio</em> method.</p>
<ol style="list-style-type: decimal">
<li>Install and load required packages</li>
<li>Download DLPFC snRNA-seq data</li>
<li>Find MeanRatio marker genes with
<code><a href="../reference/get_mean_ratio.html">DeconvoBuddies::get_mean_ratio()</a></code>
</li>
<li>Find 1vALL marker genes with
<code>DeconvoBuddies::findMarkers_1vALL()</code>
</li>
<li>Compare marker gene selection</li>
<li>Visualize marker genes expression with
<code>DeconcoBuddies::plot_gene_express()</code> and related
functions</li>
</ol>
</div>
</div>
<div class="section level2">
<h2 id="install-and-load-required-packages">1. Install and load required packages<a class="anchor" aria-label="anchor" href="#install-and-load-required-packages"></a>
</h2>
<p><code>R</code> is an open-source statistical environment which can be
easily modified to enhance its functionality via packages. <em><a href="https://bioconductor.org/packages/3.20/DeconvoBuddies" class="external-link">DeconvoBuddies</a></em>
is a <code>R</code> package available via the <a href="http://bioconductor.org" class="external-link">Bioconductor</a> repository for packages.
<code>R</code> can be installed on any operating system from <a href="https://cran.r-project.org/" class="external-link">CRAN</a> after which you can install
<em><a href="https://bioconductor.org/packages/3.20/DeconvoBuddies" class="external-link">DeconvoBuddies</a></em>
by using the following commands in your <code>R</code> session:</p>
<div class="section level3">
<h3 id="install-deconvobuddies">Install <code>DeconvoBuddies</code><a class="anchor" aria-label="anchor" href="#install-deconvobuddies"></a>
</h3>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"DeconvoBuddies"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Check that you have a valid Bioconductor installation</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/valid.html" class="external-link">valid</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="load-other-packages">Load Other Packages<a class="anchor" aria-label="anchor" href="#load-other-packages"></a>
</h3>
<p>Let’s load the packages will use in this vignette.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Packages for different types of RNA-seq data structures in R</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"SingleCellExperiment"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## For downloading data</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/LieberInstitute/spatialLIBD" class="external-link">"spatialLIBD"</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Other helper packages for this vignette</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://dplyr.tidyverse.org" class="external-link">"dplyr"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://ggplot2.tidyverse.org" class="external-link">"ggplot2"</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Our main package</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/lahuuki/DeconvoBuddies" class="external-link">"DeconvoBuddies"</a></span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="download-dlpfc-snrna-seq-data-">2. Download DLPFC snRNA-seq data.<a class="anchor" aria-label="anchor" href="#download-dlpfc-snrna-seq-data-"></a>
</h2>
<p>Here we will download single nucleus RNA-seq data from the Human
DLPFC with 77k nuclei x 36k genes <span class="citation">(Huuki-Myers et
al. 2024)</span>. This data is stored in a
<code>SingleCellExperiment</code> object. The nuclei in this dataset are
labeled by cell types at a few resolutions, we will focus on the “broad”
resolution that contains seven cell types.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Use spatialLIBD to fetch the snRNA-seq dataset</span></span>
<span><span class="va">sce_path_zip</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/fetch_deconvo_data.html">fetch_deconvo_data</a></span><span class="op">(</span><span class="st">"sce"</span><span class="op">)</span></span>
<span><span class="co">#&gt; 2024-10-18 19:06:03.235525 loading file /github/home/.cache/R/BiocFileCache/39a7e83491b_sce_DLPFC_annotated.zip%3Fdl%3D1</span></span>
<span></span>
<span><span class="co">## unzip and load the data</span></span>
<span><span class="va">sce_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/unzip.html" class="external-link">unzip</a></span><span class="op">(</span><span class="va">sce_path_zip</span>, exdir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu">HDF5Array</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/HDF5Array/man/saveHDF5SummarizedExperiment.html" class="external-link">loadHDF5SummarizedExperiment</a></span><span class="op">(</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>, <span class="st">"sce_DLPFC_annotated"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># lobstr::obj_size(sce)</span></span>
<span><span class="co"># 172.28 MB</span></span>
<span></span>
<span><span class="co">## exclude Ambiguous cell type</span></span>
<span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="va">sce</span><span class="op">[</span>, <span class="va">sce</span><span class="op">$</span><span class="va">cellType_broad_hc</span> <span class="op">!=</span> <span class="st">"Ambiguous"</span><span class="op">]</span></span>
<span><span class="va">sce</span><span class="op">$</span><span class="va">cellType_broad_hc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/droplevels.html" class="external-link">droplevels</a></span><span class="op">(</span><span class="va">sce</span><span class="op">$</span><span class="va">cellType_broad_hc</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Check the broad cell type distribution</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">sce</span><span class="op">$</span><span class="va">cellType_broad_hc</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     Astro EndoMural     Micro     Oligo       OPC     Excit     Inhib </span></span>
<span><span class="co">#&gt;      3979      2157      1601     10894      1940     24809     11067</span></span>
<span></span>
<span><span class="co">## We're going to subset to the first 5k genes to save memory</span></span>
<span><span class="co">## In a real application you'll want to use the full dataset</span></span>
<span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="va">sce</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fl">5000</span><span class="op">)</span>, <span class="op">]</span></span>
<span></span>
<span><span class="co">## check the final dimensions of the dataset</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1]  5000 56447</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="find-meanratio-marker-genes">3. Find MeanRatio marker genes<a class="anchor" aria-label="anchor" href="#find-meanratio-marker-genes"></a>
</h2>
<p>To find <em>Mean Ratio</em> marker genes for the data in
<code>sce</code> we’ll use the function
<code><a href="../reference/get_mean_ratio.html">DeconvoBuddies::get_mean_ratio()</a></code>, this function takes a
<code>SingleCellExperiment</code> object <code>sce</code> the name of
the column in the <code>colData(sce)</code> that contains the cell type
annotations of interest (here we’ll use <code>cellType_broad_hc</code>),
and optionally you can also supply additional column names from the
<code>rowData(sce)</code> to add the <code>gene_name</code> and/or
<code>gene_ensembl</code> information to the table output of
<code>get_mean_ratio</code>.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># calculate the Mean Ratio of genes for each cell type in sce</span></span>
<span><span class="va">marker_stats_MeanRatio</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_mean_ratio.html">get_mean_ratio</a></span><span class="op">(</span></span>
<span>    sce <span class="op">=</span> <span class="va">sce</span>, <span class="co"># sce is the SingleCellExperiment with our data</span></span>
<span>    assay_name <span class="op">=</span> <span class="st">"logcounts"</span>, <span class="co">## assay to use, we recommend logcounts [default]</span></span>
<span>    cellType_col <span class="op">=</span> <span class="st">"cellType_broad_hc"</span>, <span class="co"># column in colData with cell type info</span></span>
<span>    gene_ensembl <span class="op">=</span> <span class="st">"gene_id"</span>, <span class="co"># column in rowData with ensembl gene ids</span></span>
<span>    gene_name <span class="op">=</span> <span class="st">"gene_name"</span> <span class="co"># column in rowData with gene names/symbols</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>The function <code><a href="../reference/get_mean_ratio.html">get_mean_ratio()</a></code> returns a
<code>tibble</code> with the following columns:</p>
<ul>
<li>
<code>gene</code> is the name of the gene (from
rownames(<code>sce</code>)).</li>
<li>
<code>cellType.target</code> is the cell type we’re finding marker
genes for.</li>
<li>
<code>mean.target</code> is the mean expression of <code>gene</code>
for <code>cellType.target</code>.</li>
<li>
<code>cellType.2nd</code> is the second highest non-target cell
type.</li>
<li>
<code>mean.2nd</code> is the mean expression of <code>gene</code>
for <code>cellType.2nd</code>.</li>
<li>
<code>MeanRatio</code> is the ratio of
<code>mean.target/mean.2nd</code>.</li>
<li>
<code>MeanRatio.rank</code> is the rank of <code>MeanRatio</code>
for the cell type.</li>
<li>
<code>MeanRatio.anno</code> is an annotation of the
<code>MeanRatio</code> calculation helpful for plotting.</li>
<li>
<code>gene_ensembl</code> &amp; <code>gene_name</code> optional cols
from <code>rowData(sce)</code> specified by the user to add gene
information</li>
</ul>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Explore the tibble output</span></span>
<span><span class="va">marker_stats_MeanRatio</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 1,721 × 10</span></span></span>
<span><span class="co">#&gt;    gene     cellType.target mean.target cellType.2nd mean.2nd MeanRatio</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>                 <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> LYPD6B   Inhib                 1.20  Excit          0.096<span style="text-decoration: underline;">7</span>     12.4 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> GREM2    Inhib                 1.20  Excit          0.271       4.44</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> IGSF3    Inhib                 0.899 Excit          0.241       3.72</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> GALNT14  Inhib                 1.86  Excit          0.503       3.69</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> LYPD6    Inhib                 1.09  Astro          0.435       2.51</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> SLC35D1  Inhib                 0.781 OPC            0.375       2.08</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> FLVCR1   Inhib                 0.576 Excit          0.336       1.72</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> RAVER2   Inhib                 1.09  OPC            0.651       1.68</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> ARHGEF11 Inhib                 1.81  Excit          1.10        1.64</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> VAV3     Inhib                 1.68  Astro          1.03        1.63</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 1,711 more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 4 more variables: MeanRatio.rank &lt;int&gt;, MeanRatio.anno &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   gene_ensembl &lt;chr&gt;, gene_name &lt;chr&gt;</span></span></span>
<span></span>
<span><span class="co">## genes with the highest MeanRatio are the best marker genes for each cell type</span></span>
<span><span class="va">marker_stats_MeanRatio</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">MeanRatio.rank</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 7 × 10</span></span></span>
<span><span class="co">#&gt;   gene       cellType.target mean.target cellType.2nd mean.2nd MeanRatio</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>                 <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> LYPD6B     Inhib                  1.20 Excit          0.096<span style="text-decoration: underline;">7</span>     12.4 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> AC012494.1 Oligo                  2.37 OPC            0.147      16.1 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> MIR3681HG  OPC                    1.58 Excit          0.205       7.69</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> AC011995.2 Excit                  1.01 Inhib          0.135       7.51</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> PRDM16     Astro                  1.97 EndoMural      0.142      13.9 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> SLC2A1     EndoMural              1.49 Astro          0.146      10.2 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">7</span> LINC01141  Micro                  1.57 Excit          0.064<span style="text-decoration: underline;">0</span>     24.5 </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 4 more variables: MeanRatio.rank &lt;int&gt;, MeanRatio.anno &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   gene_ensembl &lt;chr&gt;, gene_name &lt;chr&gt;</span></span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="find-1vall-marker-genes">4. Find 1vALL marker genes<a class="anchor" aria-label="anchor" href="#find-1vall-marker-genes"></a>
</h2>
<p>To further explore cell type marker genes it can be helpful to also
calculate the 1vALL stats for the dataset. To help with this we have
included the function <code>DeconvoBuddies::findMarkers_1vALL()</code>,
which is a wrapper for <code><a href="https://rdrr.io/pkg/scran/man/findMarkers.html" class="external-link">scran::findMarkers()</a></code> that iterates
through cell types and creates an table output in a compatible with the
output <code><a href="../reference/get_mean_ratio.html">DeconvoBuddies::get_mean_ratio()</a></code>.</p>
<p>Similarity to <code>get_mean_ratio</code> this function requires the
<code>sce</code> object, <code>cellType_col</code> to define cell types,
and <code>assay_name</code>. But <code>findMarkers_1vALL()</code> also
takes a model (<code>mod</code>) to use as design in
<code><a href="https://rdrr.io/pkg/scran/man/findMarkers.html" class="external-link">scran::findMarkers()</a></code>, in this example we will control for
donor which is stored as <code>BrNum</code>.</p>
<p>Note this function can take a bit of time to run.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Run 1vALL DE to find markers for each cell type</span></span>
<span><span class="va">marker_stats_1vAll</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/findMarkers_1vAll.html">findMarkers_1vAll</a></span><span class="op">(</span></span>
<span>    sce <span class="op">=</span> <span class="va">sce</span>, <span class="co"># sce is the SingleCellExperiment with our data</span></span>
<span>    assay_name <span class="op">=</span> <span class="st">"counts"</span>,</span>
<span>    cellType_col <span class="op">=</span> <span class="st">"cellType_broad_hc"</span>, <span class="co"># column in colData with cell type info</span></span>
<span>    mod <span class="op">=</span> <span class="st">"~BrNum"</span> <span class="co"># Control for donor stored in "BrNum" with mod</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>The function <code>findMarkers_1vALL()</code> returns a
<code>tibble</code> with the following columns:</p>
<ul>
<li>
<code>gene</code> is the name of the gene (from
rownames(<code>sce</code>)).</li>
<li>
<code>logFC</code> the log fold change from the DE test</li>
<li>
<code>log.p.value</code> the log of the p-value of the DE test</li>
<li>
<code>log.FDR</code> the log of the False Discovery Rate adjusted
p.value</li>
<li>
<code>std.logFC</code> the standard logFC</li>
<li>
<code>cellType.target</code> the cell type we’re finding marker
genes for</li>
<li>
<code>std.logFC.rank</code> the rank of <code>std.logFC</code> for
each cell type</li>
<li>
<code>std.logFC.anno</code> is an annotation of the
<code>std.logFC</code> value helpful for plotting.</li>
</ul>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Explore the tibble output</span></span>
<span><span class="va">marker_stats_1vAll</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 35,000 × 8</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># Groups:   cellType.target [7]</span></span></span>
<span><span class="co">#&gt;    gene    logFC log.p.value log.FDR std.logFC cellType.target std.logFC.rank</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>                    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> SILC1   2.77       -<span style="color: #BB0000; text-decoration: underline;">6</span><span style="color: #BB0000;">747.</span>  -<span style="color: #BB0000; text-decoration: underline;">6</span><span style="color: #BB0000;">739.</span>      1.42 Inhib                        1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> LYPD6B  2.48       -<span style="color: #BB0000; text-decoration: underline;">6</span><span style="color: #BB0000;">338.</span>  -<span style="color: #BB0000; text-decoration: underline;">6</span><span style="color: #BB0000;">330.</span>      1.37 Inhib                        2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> LYPD6   2.18       -<span style="color: #BB0000; text-decoration: underline;">6</span><span style="color: #BB0000;">147.</span>  -<span style="color: #BB0000; text-decoration: underline;">6</span><span style="color: #BB0000;">140.</span>      1.35 Inhib                        3</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> ALK     4.56       -<span style="color: #BB0000; text-decoration: underline;">5</span><span style="color: #BB0000;">707.</span>  -<span style="color: #BB0000; text-decoration: underline;">5</span><span style="color: #BB0000;">700.</span>      1.30 Inhib                        4</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> WLS     3.26       -<span style="color: #BB0000; text-decoration: underline;">5</span><span style="color: #BB0000;">111.</span>  -<span style="color: #BB0000; text-decoration: underline;">5</span><span style="color: #BB0000;">104.</span>      1.22 Inhib                        5</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> PLD5    5.85       -<span style="color: #BB0000; text-decoration: underline;">4</span><span style="color: #BB0000;">711.</span>  -<span style="color: #BB0000; text-decoration: underline;">4</span><span style="color: #BB0000;">704.</span>      1.17 Inhib                        6</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> KIRREL1 1.04       -<span style="color: #BB0000; text-decoration: underline;">4</span><span style="color: #BB0000;">539.</span>  -<span style="color: #BB0000; text-decoration: underline;">4</span><span style="color: #BB0000;">532.</span>      1.14 Inhib                        7</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> TRIM67  0.622      -<span style="color: #BB0000; text-decoration: underline;">4</span><span style="color: #BB0000;">326.</span>  -<span style="color: #BB0000; text-decoration: underline;">4</span><span style="color: #BB0000;">319.</span>      1.11 Inhib                        8</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> GREM2   2.11       -<span style="color: #BB0000; text-decoration: underline;">4</span><span style="color: #BB0000;">048.</span>  -<span style="color: #BB0000; text-decoration: underline;">4</span><span style="color: #BB0000;">042.</span>      1.07 Inhib                        9</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> IGSF3   1.27       -<span style="color: #BB0000; text-decoration: underline;">3</span><span style="color: #BB0000;">783.</span>  -<span style="color: #BB0000; text-decoration: underline;">3</span><span style="color: #BB0000;">777.</span>      1.04 Inhib                       10</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 34,990 more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 1 more variable: std.logFC.anno &lt;chr&gt;</span></span></span>
<span></span>
<span><span class="co">## genes with the highest MeanRatio are the best marker genes for each cell type</span></span>
<span><span class="va">marker_stats_1vAll</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">std.logFC.rank</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 7 × 8</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># Groups:   cellType.target [7]</span></span></span>
<span><span class="co">#&gt;   gene       logFC log.p.value log.FDR std.logFC cellType.target std.logFC.rank</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>                    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> SILC1      2.77       -<span style="color: #BB0000; text-decoration: underline;">6</span><span style="color: #BB0000;">747.</span>  -<span style="color: #BB0000; text-decoration: underline;">6</span><span style="color: #BB0000;">739.</span>      1.42 Inhib                        1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> RNF220    10.7        -<span style="color: #BB0000; text-decoration: underline;">8</span><span style="color: #BB0000;">373.</span>  -<span style="color: #BB0000; text-decoration: underline;">8</span><span style="color: #BB0000;">365.</span>      1.52 Oligo                        1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> BX284613   7.42      -<span style="color: #BB0000; text-decoration: underline;">12</span><span style="color: #BB0000;">763.</span> -<span style="color: #BB0000; text-decoration: underline;">12</span><span style="color: #BB0000;">754.</span>      4.20 OPC                          1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> KIAA1211L  7.89       -<span style="color: #BB0000; text-decoration: underline;">8</span><span style="color: #BB0000;">371.</span>  -<span style="color: #BB0000; text-decoration: underline;">8</span><span style="color: #BB0000;">362.</span>      1.23 Excit                        1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> PRDM16     1.90       -<span style="color: #BB0000; text-decoration: underline;">8</span><span style="color: #BB0000;">853.</span>  -<span style="color: #BB0000; text-decoration: underline;">8</span><span style="color: #BB0000;">844.</span>      2.38 Astro                        1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> EPAS1      6.50       -<span style="color: #BB0000; text-decoration: underline;">9</span><span style="color: #BB0000;">480.</span>  -<span style="color: #BB0000; text-decoration: underline;">9</span><span style="color: #BB0000;">471.</span>      3.31 EndoMural                    1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">7</span> CSF3R      0.808      -<span style="color: #BB0000; text-decoration: underline;">9</span><span style="color: #BB0000;">670.</span>  -<span style="color: #BB0000; text-decoration: underline;">9</span><span style="color: #BB0000;">661.</span>      3.86 Micro                        1</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 1 more variable: std.logFC.anno &lt;chr&gt;</span></span></span></code></pre></div>
<p>As this is a differential expression test, you can create volcano
plots to explore the outputs. 🌋</p>
<p>Note that with the default option “up” for direction only
up-regulated genes are considered marker candidates, so all genes with
logFC&lt;1 will have a p.value=0. As a results these plots will only
have the right half of the volcano shape. If you’d like all p-values set
<code>findMarkers_1vALL(direction="any")</code>.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Create volcano plots of DE stats from 1vALL</span></span>
<span><span class="va">marker_stats_1vAll</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">logFC</span>, <span class="op">-</span><span class="va">log.p.value</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">cellType.target</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_vline</a></span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="op">-</span><span class="fl">1</span><span class="op">)</span>, linetype <span class="op">=</span> <span class="st">"dashed"</span>, color <span class="op">=</span> <span class="st">"red"</span><span class="op">)</span></span></code></pre></div>
<p><img src="Marker_Finding_files/figure-html/volcano%20plots-1.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="compare-marker-gene-selection">5. Compare Marker Gene Selection<a class="anchor" aria-label="anchor" href="#compare-marker-gene-selection"></a>
</h2>
<p>Let’s join the two <code>marker_stats</code> tables together to
compare the findings of the two methods.</p>
<p>Note as we are using a subset of data for this example, for some
genes there is not enough data to test and 1vALL will have some missing
results.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## join the two marker_stats tables</span></span>
<span><span class="va">marker_stats</span> <span class="op">&lt;-</span> <span class="va">marker_stats_MeanRatio</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span><span class="va">marker_stats_1vAll</span>, by <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/join_by.html" class="external-link">join_by</a></span><span class="op">(</span><span class="va">gene</span>, <span class="va">cellType.target</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Check stats for our top genes</span></span>
<span><span class="va">marker_stats</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">MeanRatio.rank</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">gene</span>, <span class="va">cellType.target</span>, <span class="va">MeanRatio</span>, <span class="va">MeanRatio.rank</span>, <span class="va">std.logFC</span>, <span class="va">std.logFC.rank</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 7 × 6</span></span></span>
<span><span class="co">#&gt;   gene       cellType.target MeanRatio MeanRatio.rank std.logFC std.logFC.rank</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>               <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> LYPD6B     Inhib               12.4               1      1.37              2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> AC012494.1 Oligo               16.1               1     <span style="color: #BB0000;">NA</span>                <span style="color: #BB0000;">NA</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> MIR3681HG  OPC                  7.69              1      1.42             11</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> AC011995.2 Excit                7.51              1     <span style="color: #BB0000;">NA</span>                <span style="color: #BB0000;">NA</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> PRDM16     Astro               13.9               1      2.38              1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> SLC2A1     EndoMural           10.2               1      2.30              4</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">7</span> LINC01141  Micro               24.5               1      2.07              8</span></span></code></pre></div>
<div class="section level3">
<h3 id="hockey-stick-plots">Hockey Stick Plots<a class="anchor" aria-label="anchor" href="#hockey-stick-plots"></a>
</h3>
<p>Plotting the values of Mean Ratio vs. standard log fold change (from
1vAll) we create what we call “hockey stick plots” 🏒. These plots help
visualize the distribution of <code>MeanRatio</code> and standard
<code>logFC</code> values.</p>
<p align="center">
</p>
<div class="float">
<img src="http://research.libd.org/DeconvoBuddies/reference/figures/hockey_plot_anno.png" alt="Figure 2. Hockey Stick plots compare values of 1vALL logFC vs. MeanRatio and help select marker genes."><div class="figcaption">Figure 2. Hockey Stick plots compare values of
<em>1vALL</em> logFC vs. <em>MeanRatio</em> and help select marker
genes.</div>
</div>

<p>Typically for a cell type see most genes have low Mean Ratio and low
fold change, these genes are not marker genes (red box in illustration
above on the bottom left side).</p>
<p>Genes with higher fold change from 1vALL are better marker gene
candidates, but most have low <em>MeanRatio</em> values indicating that
one or more non-target cell types have high expression for that gene,
causing noise (orange box on the top left side).</p>
<p>Genes with high M<em>eanRatio</em> typically also have high
<em>1vALL</em> standard fold changes, these are cell type specific
marker genes we are selecting for (green box on the top right side).</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># create hockey stick plots to compare MeanRatio and standard logFC values.</span></span>
<span><span class="va">marker_stats</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">MeanRatio</span>, <span class="va">std.logFC</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">cellType.target</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Removed 29 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">#&gt; (`geom_point()`).</span></span></code></pre></div>
<p><img src="Marker_Finding_files/figure-html/hockey%20stick%20plots-1.png" width="700"></p>
<p>We can see a “hockey stick” shape in most of the cell types, with a
few marker genes with high values for both <code>logFC</code> and
<code>MeanRatio</code>.</p>
</div>
</div>
<div class="section level2">
<h2 id="visualize-marker-genes-expression">6. Visualize Marker Genes Expression<a class="anchor" aria-label="anchor" href="#visualize-marker-genes-expression"></a>
</h2>
<p>An important step for ensuring you have selected high quality marker
genes is to visualize their expression over the cell types in the
dataset. <code>DeconvoBuddies</code> has several functions to help
quickly plot gene expression at a few levels:</p>
<p><code><a href="../reference/plot_gene_express.html">plot_gene_express()</a></code> plots the expression of one or more
genes as a violin plot.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## plot expression of two genes from a list</span></span>
<span><span class="fu"><a href="../reference/plot_gene_express.html">plot_gene_express</a></span><span class="op">(</span></span>
<span>    sce <span class="op">=</span> <span class="va">sce</span>,</span>
<span>    category <span class="op">=</span> <span class="st">"cellType_broad_hc"</span>,</span>
<span>    genes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"SLC2A1"</span>, <span class="st">"CREG2"</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="Marker_Finding_files/figure-html/plot_gene_express-1.png" width="700"></p>
<p><code><a href="../reference/plot_marker_express.html">plot_marker_express()</a></code> plots the top n marker genes for a
specified cell type based on the values from <code>marker_stats</code>.
Annotations for the details of the <em>MeanRatio</em> value +
calculation are added to each panel.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># plot the top 10 MeanRatio genes for Excit</span></span>
<span><span class="fu"><a href="../reference/plot_marker_express.html">plot_marker_express</a></span><span class="op">(</span></span>
<span>    sce <span class="op">=</span> <span class="va">sce</span>,</span>
<span>    stats <span class="op">=</span> <span class="va">marker_stats</span>,</span>
<span>    cell_type <span class="op">=</span> <span class="st">"Excit"</span>,</span>
<span>    n_genes <span class="op">=</span> <span class="fl">10</span>,</span>
<span>    cellType_col <span class="op">=</span> <span class="st">"cellType_broad_hc"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="Marker_Finding_files/figure-html/plot_marker_express%20MeanRatio-1.png" width="700"></p>
<p>In these violin plots we can see these genes have high expression in
the target cell type Excit , and mostly low expression nuclei in the
other cell types, sometimes even no expression.</p>
<p>This function defaults to selecting genes by the <em>MeanRatio</em>
stats, but can also be used to plot the <em>1vAll</em> genes.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># plot the top 10 1vAll genes for Excit</span></span>
<span><span class="fu"><a href="../reference/plot_marker_express.html">plot_marker_express</a></span><span class="op">(</span></span>
<span>    sce <span class="op">=</span> <span class="va">sce</span>,</span>
<span>    stats <span class="op">=</span> <span class="va">marker_stats</span>,</span>
<span>    cell_type <span class="op">=</span> <span class="st">"Excit"</span>,</span>
<span>    n_genes <span class="op">=</span> <span class="fl">10</span>,</span>
<span>    rank_col <span class="op">=</span> <span class="st">"std.logFC.rank"</span>, <span class="co">## use logFC cols from 1vALL</span></span>
<span>    anno_col <span class="op">=</span> <span class="st">"std.logFC.anno"</span>,</span>
<span>    cellType_col <span class="op">=</span> <span class="st">"cellType_broad_hc"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="Marker_Finding_files/figure-html/plot_marker_express%20logFC-1.png" width="700"></p>
<p>We can see in the top <em>1vALL</em> genes there is some expression
of these genes in Inhib nuclei in addition to the target cell type
Excit.</p>
<p><code><a href="../reference/plot_marker_express_ALL.html">plot_marker_express_ALL()</a></code> plots the top marker genes for
all cell types. This is a quick and easy way to look at the the top
markers in your dataset, which is an important step and can help
identify genes with multimodal distributions that may confound the
MeanRatio method.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># plot the top 10 1vAll genes for all cell types</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="../reference/plot_marker_express_ALL.html">plot_marker_express_ALL</a></span><span class="op">(</span></span>
<span>    sce <span class="op">=</span> <span class="va">sce</span>,</span>
<span>    stats <span class="op">=</span> <span class="va">marker_stats</span>,</span>
<span>    n_genes <span class="op">=</span> <span class="fl">10</span>,</span>
<span>    cellType_col <span class="op">=</span> <span class="st">"cellType_broad_hc"</span></span>
<span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [[1]]</span></span></code></pre></div>
<p><img src="Marker_Finding_files/figure-html/plot_marker_express_ALL-1.png" width="700"></p>
<pre><code><span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [[2]]</span></span></code></pre>
<p><img src="Marker_Finding_files/figure-html/plot_marker_express_ALL-2.png" width="700"></p>
<pre><code><span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [[3]]</span></span></code></pre>
<p><img src="Marker_Finding_files/figure-html/plot_marker_express_ALL-3.png" width="700"></p>
<pre><code><span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [[4]]</span></span></code></pre>
<p><img src="Marker_Finding_files/figure-html/plot_marker_express_ALL-4.png" width="700"></p>
<pre><code><span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [[5]]</span></span></code></pre>
<p><img src="Marker_Finding_files/figure-html/plot_marker_express_ALL-5.png" width="700"></p>
<pre><code><span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [[6]]</span></span></code></pre>
<p><img src="Marker_Finding_files/figure-html/plot_marker_express_ALL-6.png" width="700"></p>
<pre><code><span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [[7]]</span></span></code></pre>
<p><img src="Marker_Finding_files/figure-html/plot_marker_express_ALL-7.png" width="700"></p>
<p>The violin plots can also be directly printed to a PDF file using the
built in argument
<code>plot_marker_express_ALL(pdf = "my_marker_genes.pdf")</code> for
portability and easy sharing.</p>
</div>
<div class="section level2">
<h2 id="summary">Summary<a class="anchor" aria-label="anchor" href="#summary"></a>
</h2>
<p>In this vignette we covered the importance of finding marker genes,
and introduced our method for finding cell type specific genes
<em>MeanRatio</em>. We covered how to find and compare
<em>MeanRatio</em> marker genes with <code><a href="../reference/get_mean_ratio.html">get_mean_ratio()</a></code>, and
<em>1vALL</em> marker genes with <code>findMarkers_1vALL()</code>. And
finally, how to visualize the expression of these marker genes with
<code><a href="../reference/plot_marker_express.html">plot_marker_express()</a></code> and related functions.</p>
<p><strong>Next Steps</strong></p>
<p>For an example of how to use these marker genes in a deconvolution
workflow, check out <a href="https://research.libd.org/DeconvoBuddies/articles/Deconvolution_Benchmark_DLPFC.html" class="external-link">Vignette:
Deconvolution Benchmark in Human DLPFC</a>.</p>
<p>We hope this vignette and <em>DeconvoBuddies</em> helps you with your
research goals! Thanks for reading 😁</p>
</div>
<div class="section level2">
<h2 id="reproducibility">Reproducibility<a class="anchor" aria-label="anchor" href="#reproducibility"></a>
</h2>
<p>The <em><a href="https://bioconductor.org/packages/3.20/DeconvoBuddies" class="external-link">DeconvoBuddies</a></em>
package (Huuki-Myers, Maynard, Hicks, Zandi, Kleinman, Hyde, Goes, and
Collado-Torres, 2024) was made possible thanks to:</p>
<ul>
<li>R (R Core Team, 2024)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.20/BiocStyle" class="external-link">BiocStyle</a></em>
(Oleś, 2024)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=knitr" class="external-link">knitr</a></em> (Xie,
2024)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=RefManageR" class="external-link">RefManageR</a></em>
(McLean, 2017)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=rmarkdown" class="external-link">rmarkdown</a></em>
(Allaire, Xie, Dervieux, McPherson, Luraschi, Ushey, Atkins, Wickham,
Cheng, Chang, and Iannone, 2024)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=sessioninfo" class="external-link">sessioninfo</a></em>
(Wickham, Chang, Flight, Müller, and Hester, 2021)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=testthat" class="external-link">testthat</a></em>
(Wickham, 2011)</li>
</ul>
<p>This package was developed using <em><a href="https://bioconductor.org/packages/3.20/biocthis" class="external-link">biocthis</a></em>.</p>
<p>Code for creating the vignette</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Create the vignette</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/rstudio/rmarkdown" class="external-link">"rmarkdown"</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/system.time.html" class="external-link">system.time</a></span><span class="op">(</span><span class="fu"><a href="https://pkgs.rstudio.com/rmarkdown/reference/render.html" class="external-link">render</a></span><span class="op">(</span><span class="st">"Marker_Finding.Rmd"</span>, <span class="st">"BiocStyle::html_document"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Extract the R code</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://yihui.org/knitr/" class="external-link">"knitr"</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/knit.html" class="external-link">knit</a></span><span class="op">(</span><span class="st">"Marker_Finding.Rmd"</span>, tangle <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>Date the vignette was generated.</p>
<pre><code><span><span class="co">#&gt; [1] "2024-10-18 19:08:14 UTC"</span></span></code></pre>
<p>Wallclock time spent generating the vignette.</p>
<pre><code><span><span class="co">#&gt; Time difference of 2.398 mins</span></span></code></pre>
<p><code>R</code> session information.</p>
<pre><code><span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt;  <span style="color: #555555; font-style: italic;">setting </span> <span style="color: #555555; font-style: italic;">value</span></span></span>
<span><span class="co">#&gt;  version  R version 4.4.1 (2024-06-14)</span></span>
<span><span class="co">#&gt;  os       Ubuntu 22.04.5 LTS</span></span>
<span><span class="co">#&gt;  system   x86_64, linux-gnu</span></span>
<span><span class="co">#&gt;  ui       X11</span></span>
<span><span class="co">#&gt;  language en</span></span>
<span><span class="co">#&gt;  collate  en_US.UTF-8</span></span>
<span><span class="co">#&gt;  ctype    en_US.UTF-8</span></span>
<span><span class="co">#&gt;  tz       UTC</span></span>
<span><span class="co">#&gt;  date     2024-10-18</span></span>
<span><span class="co">#&gt;  pandoc   3.4 @ /usr/bin/ (via rmarkdown)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt;  <span style="color: #555555; font-style: italic;">package             </span> <span style="color: #555555; font-style: italic;">*</span> <span style="color: #555555; font-style: italic;">version  </span> <span style="color: #555555; font-style: italic;">date (UTC)</span> <span style="color: #555555; font-style: italic;">lib</span> <span style="color: #555555; font-style: italic;">source</span></span></span>
<span><span class="co">#&gt;  abind                  1.4-8     <span style="color: #555555;">2024-09-12</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  AnnotationDbi          1.67.0    <span style="color: #555555;">2024-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  AnnotationHub          3.13.3    <span style="color: #555555;">2024-08-19</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  attempt                0.3.1     <span style="color: #555555;">2020-05-03</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  backports              1.5.0     <span style="color: #555555;">2024-05-23</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  beachmat               2.21.8    <span style="color: #555555;">2024-10-17</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  beeswarm               0.4.0     <span style="color: #555555;">2021-06-01</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  benchmarkme            1.0.8     <span style="color: #555555;">2022-06-12</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  benchmarkmeData        1.0.4     <span style="color: #555555;">2020-04-23</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  bibtex                 0.5.1     <span style="color: #555555;">2023-01-26</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  Biobase              * 2.65.1    <span style="color: #555555;">2024-08-28</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  BiocFileCache          2.13.2    <span style="color: #555555;">2024-10-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  BiocGenerics         * 0.51.3    <span style="color: #555555;">2024-10-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  BiocIO                 1.15.2    <span style="color: #555555;">2024-08-23</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  BiocManager            1.30.25   <span style="color: #555555;">2024-08-28</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  BiocNeighbors          1.99.2    <span style="color: #555555;">2024-10-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  BiocParallel           1.39.0    <span style="color: #555555;">2024-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  BiocSingular           1.21.4    <span style="color: #555555;">2024-09-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  BiocStyle            * 2.33.1    <span style="color: #555555;">2024-06-12</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  BiocVersion            3.20.0    <span style="color: #555555;">2024-05-01</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  Biostrings             2.73.2    <span style="color: #555555;">2024-09-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  bit                    4.5.0     <span style="color: #555555;">2024-09-20</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  bit64                  4.5.2     <span style="color: #555555;">2024-09-22</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  bitops                 1.0-9     <span style="color: #555555;">2024-10-03</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  blob                   1.2.4     <span style="color: #555555;">2023-03-17</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  bluster                1.15.1    <span style="color: #555555;">2024-09-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  bookdown               0.41      <span style="color: #555555;">2024-10-16</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  bslib                  0.8.0     <span style="color: #555555;">2024-07-29</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  cachem                 1.1.0     <span style="color: #555555;">2024-05-16</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  cli                    3.6.3     <span style="color: #555555;">2024-06-21</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  cluster                2.1.6     <span style="color: #555555;">2023-12-01</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  codetools              0.2-20    <span style="color: #555555;">2024-03-31</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  colorspace             2.1-1     <span style="color: #555555;">2024-07-26</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  config                 0.3.2     <span style="color: #555555;">2023-08-30</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  cowplot                1.1.3     <span style="color: #555555;">2024-01-22</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  crayon                 1.5.3     <span style="color: #555555;">2024-06-20</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  curl                   5.2.3     <span style="color: #555555;">2024-09-20</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  data.table             1.16.2    <span style="color: #555555;">2024-10-10</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  DBI                    1.2.3     <span style="color: #555555;">2024-06-02</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  dbplyr                 2.5.0     <span style="color: #555555;">2024-03-19</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  DeconvoBuddies       * 0.99.3    <span style="color: #555555;">2024-10-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  DelayedArray           0.31.14   <span style="color: #555555;">2024-10-03</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  desc                   1.4.3     <span style="color: #555555;">2023-12-10</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  digest                 0.6.37    <span style="color: #555555;">2024-08-19</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  doParallel             1.0.17    <span style="color: #555555;">2022-02-07</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  dotCall64              1.2       <span style="color: #555555;">2024-10-04</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  dplyr                * 1.1.4     <span style="color: #555555;">2023-11-17</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  dqrng                  0.4.1     <span style="color: #555555;">2024-05-28</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  DT                     0.33      <span style="color: #555555;">2024-04-04</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  edgeR                  4.3.19    <span style="color: #555555;">2024-10-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  evaluate               1.0.1     <span style="color: #555555;">2024-10-10</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  ExperimentHub          2.13.1    <span style="color: #555555;">2024-07-31</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  fansi                  1.0.6     <span style="color: #555555;">2023-12-08</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  farver                 2.1.2     <span style="color: #555555;">2024-05-13</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  fastmap                1.2.0     <span style="color: #555555;">2024-05-15</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  fields                 16.3      <span style="color: #555555;">2024-09-30</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  filelock               1.0.3     <span style="color: #555555;">2023-12-11</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  foreach                1.5.2     <span style="color: #555555;">2022-02-02</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  fs                     1.6.4     <span style="color: #555555;">2024-04-25</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  generics               0.1.3     <span style="color: #555555;">2022-07-05</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  GenomeInfoDb         * 1.41.2    <span style="color: #555555;">2024-10-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  GenomeInfoDbData       1.2.13    <span style="color: #555555;">2024-10-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  GenomicAlignments      1.41.0    <span style="color: #555555;">2024-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  GenomicRanges        * 1.57.2    <span style="color: #555555;">2024-10-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  ggbeeswarm             0.7.2     <span style="color: #555555;">2023-04-29</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  ggplot2              * 3.5.1     <span style="color: #555555;">2024-04-23</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  ggrepel                0.9.6     <span style="color: #555555;">2024-09-07</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  glue                   1.8.0     <span style="color: #555555;">2024-09-30</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  golem                  0.5.1     <span style="color: #555555;">2024-08-27</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  gridExtra              2.3       <span style="color: #555555;">2017-09-09</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  gtable                 0.3.5     <span style="color: #555555;">2024-04-22</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  HDF5Array              1.33.8    <span style="color: #555555;">2024-10-04</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  highr                  0.11      <span style="color: #555555;">2024-05-26</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  htmltools              0.5.8.1   <span style="color: #555555;">2024-04-04</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  htmlwidgets            1.6.4     <span style="color: #555555;">2023-12-06</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  httpuv                 1.6.15    <span style="color: #555555;">2024-03-26</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  httr                   1.4.7     <span style="color: #555555;">2023-08-15</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  igraph                 2.0.3     <span style="color: #555555;">2024-03-13</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  IRanges              * 2.39.2    <span style="color: #555555;">2024-07-17</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  irlba                  2.3.5.1   <span style="color: #555555;">2022-10-03</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  iterators              1.0.14    <span style="color: #555555;">2022-02-05</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  jquerylib              0.1.4     <span style="color: #555555;">2021-04-26</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  jsonlite               1.8.9     <span style="color: #555555;">2024-09-20</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  KEGGREST               1.45.1    <span style="color: #555555;">2024-06-17</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  knitr                  1.48      <span style="color: #555555;">2024-07-07</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  labeling               0.4.3     <span style="color: #555555;">2023-08-29</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  later                  1.3.2     <span style="color: #555555;">2023-12-06</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  lattice                0.22-6    <span style="color: #555555;">2024-03-20</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  lazyeval               0.2.2     <span style="color: #555555;">2019-03-15</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  lifecycle              1.0.4     <span style="color: #555555;">2023-11-07</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  limma                  3.61.12   <span style="color: #555555;">2024-09-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  locfit                 1.5-9.10  <span style="color: #555555;">2024-06-24</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  lubridate              1.9.3     <span style="color: #555555;">2023-09-27</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  magick                 2.8.5     <span style="color: #555555;">2024-09-20</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  magrittr               2.0.3     <span style="color: #555555;">2022-03-30</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  maps                   3.4.2     <span style="color: #555555;">2023-12-15</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  Matrix                 1.7-0     <span style="color: #555555;">2024-04-26</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  MatrixGenerics       * 1.17.0    <span style="color: #555555;">2024-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  matrixStats          * 1.4.1     <span style="color: #555555;">2024-09-08</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  memoise                2.0.1     <span style="color: #555555;">2021-11-26</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  metapod                1.13.0    <span style="color: #555555;">2024-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  mime                   0.12      <span style="color: #555555;">2021-09-28</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  munsell                0.5.1     <span style="color: #555555;">2024-04-01</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  paletteer              1.6.0     <span style="color: #555555;">2024-01-21</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  pillar                 1.9.0     <span style="color: #555555;">2023-03-22</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  pkgconfig              2.0.3     <span style="color: #555555;">2019-09-22</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  pkgdown                2.1.1     <span style="color: #555555;">2024-09-17</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  plotly                 4.10.4    <span style="color: #555555;">2024-01-13</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  plyr                   1.8.9     <span style="color: #555555;">2023-10-02</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  png                    0.1-8     <span style="color: #555555;">2022-11-29</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  promises               1.3.0     <span style="color: #555555;">2024-04-05</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  purrr                  1.0.2     <span style="color: #555555;">2023-08-10</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  R6                     2.5.1     <span style="color: #555555;">2021-08-19</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  rafalib                1.0.0     <span style="color: #555555;">2015-08-09</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  ragg                   1.3.3     <span style="color: #555555;">2024-09-11</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  rappdirs               0.3.3     <span style="color: #555555;">2021-01-31</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  RColorBrewer           1.1-3     <span style="color: #555555;">2022-04-03</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  Rcpp                   1.0.13    <span style="color: #555555;">2024-07-17</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  RCurl                  1.98-1.16 <span style="color: #555555;">2024-07-11</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  RefManageR           * 1.4.0     <span style="color: #555555;">2022-09-30</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  rematch2               2.1.2     <span style="color: #555555;">2020-05-01</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  reshape2               1.4.4     <span style="color: #555555;">2020-04-09</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  restfulr               0.0.15    <span style="color: #555555;">2022-06-16</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  rhdf5                  2.49.0    <span style="color: #555555;">2024-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  rhdf5filters           1.17.0    <span style="color: #555555;">2024-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  Rhdf5lib               1.27.0    <span style="color: #555555;">2024-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  rjson                  0.2.23    <span style="color: #555555;">2024-09-16</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  rlang                  1.1.4     <span style="color: #555555;">2024-06-04</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  rmarkdown              2.28      <span style="color: #555555;">2024-08-17</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  Rsamtools              2.21.2    <span style="color: #555555;">2024-09-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  RSQLite                2.3.7     <span style="color: #555555;">2024-05-27</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  rsvd                   1.0.5     <span style="color: #555555;">2021-04-16</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  rtracklayer            1.65.0    <span style="color: #555555;">2024-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  S4Arrays               1.5.11    <span style="color: #555555;">2024-10-14</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  S4Vectors            * 0.43.2    <span style="color: #555555;">2024-07-17</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  sass                   0.4.9     <span style="color: #555555;">2024-03-15</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  ScaledMatrix           1.13.0    <span style="color: #555555;">2024-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  scales                 1.3.0     <span style="color: #555555;">2023-11-28</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  scater                 1.33.4    <span style="color: #555555;">2024-07-21</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  scran                  1.33.2    <span style="color: #555555;">2024-09-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  scuttle                1.15.4    <span style="color: #555555;">2024-08-14</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  sessioninfo          * 1.2.2     <span style="color: #555555;">2021-12-06</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  shiny                  1.9.1     <span style="color: #555555;">2024-08-01</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  shinyWidgets           0.8.7     <span style="color: #555555;">2024-09-23</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  SingleCellExperiment * 1.27.2    <span style="color: #555555;">2024-05-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  spam                   2.11-0    <span style="color: #555555;">2024-10-03</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  SparseArray            1.5.45    <span style="color: #555555;">2024-10-17</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  SpatialExperiment    * 1.15.1    <span style="color: #555555;">2024-06-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  spatialLIBD          * 1.17.8    <span style="color: #555555;">2024-07-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  statmod                1.5.0     <span style="color: #555555;">2023-01-06</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  stringi                1.8.4     <span style="color: #555555;">2024-05-06</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  stringr                1.5.1     <span style="color: #555555;">2023-11-14</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  SummarizedExperiment * 1.35.4    <span style="color: #555555;">2024-10-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  systemfonts            1.1.0     <span style="color: #555555;">2024-05-15</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  textshaping            0.4.0     <span style="color: #555555;">2024-05-24</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  tibble                 3.2.1     <span style="color: #555555;">2023-03-20</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  tidyr                  1.3.1     <span style="color: #555555;">2024-01-24</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  tidyselect             1.2.1     <span style="color: #555555;">2024-03-11</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  timechange             0.3.0     <span style="color: #555555;">2024-01-18</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  UCSC.utils             1.1.0     <span style="color: #555555;">2024-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  utf8                   1.2.4     <span style="color: #555555;">2023-10-22</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  vctrs                  0.6.5     <span style="color: #555555;">2023-12-01</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  vipor                  0.4.7     <span style="color: #555555;">2023-12-18</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  viridis                0.6.5     <span style="color: #555555;">2024-01-29</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  viridisLite            0.4.2     <span style="color: #555555;">2023-05-02</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  withr                  3.0.1     <span style="color: #555555;">2024-07-31</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  xfun                   0.48      <span style="color: #555555;">2024-10-03</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  XML                    3.99-0.17 <span style="color: #555555;">2024-06-25</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  xml2                   1.3.6     <span style="color: #555555;">2023-12-04</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  xtable                 1.8-4     <span style="color: #555555;">2019-04-21</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  XVector                0.45.0    <span style="color: #555555;">2024-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  yaml                   2.3.10    <span style="color: #555555;">2024-07-26</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  zlibbioc               1.51.1    <span style="color: #555555;">2024-06-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [1] /__w/_temp/Library</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [2] /usr/local/lib/R/site-library</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [3] /usr/local/lib/R/library</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></span></code></pre>
</div>
<div class="section level2">
<h2 id="bibliography">Bibliography<a class="anchor" aria-label="anchor" href="#bibliography"></a>
</h2>
<p>This vignette was generated using <em><a href="https://bioconductor.org/packages/3.20/BiocStyle" class="external-link">BiocStyle</a></em>
(Oleś, 2024) with <em><a href="https://CRAN.R-project.org/package=knitr" class="external-link">knitr</a></em> (Xie,
2024) and <em><a href="https://CRAN.R-project.org/package=rmarkdown" class="external-link">rmarkdown</a></em>
(Allaire, Xie, Dervieux et al., 2024) running behind the scenes.</p>
<p>Citations made with <em><a href="https://CRAN.R-project.org/package=RefManageR" class="external-link">RefManageR</a></em>
(McLean, 2017).</p>
<p>
<a id="bib-allaire2024rmarkdown"></a><a href="#cite-allaire2024rmarkdown">[1]</a><cite>
J. Allaire, Y. Xie, C. Dervieux, et al. <em>rmarkdown: Dynamic Documents
for R</em>. R package version 2.28. 2024. URL:
<a href="https://github.com/rstudio/rmarkdown" class="external-link">https://github.com/rstudio/rmarkdown</a>.</cite>
</p>
<p>
<a id="bib-huukimyers2024deconvobuddies"></a><a href="#cite-huukimyers2024deconvobuddies">[2]</a><cite>
L. A. Huuki-Myers, K. R. Maynard, S. C. Hicks, et al.
<em>DeconvoBuddies: a R/Bioconductor package with deconvolution helper
functions</em>. <a href="https://github.com/LieberInstitute/DeconvoBuddies/DeconvoBuddies" class="external-link uri">https://github.com/LieberInstitute/DeconvoBuddies/DeconvoBuddies</a>
- R package version 0.99.3. 2024. DOI:
<a href="https://doi.org/10.18129/B9.bioc.DeconvoBuddies" class="external-link">10.18129/B9.bioc.DeconvoBuddies</a>.
URL:
<a href="http://www.bioconductor.org/packages/DeconvoBuddies" class="external-link">http://www.bioconductor.org/packages/DeconvoBuddies</a>.</cite>
</p>
<p>
<a id="bib-mclean2017refmanager"></a><a href="#cite-mclean2017refmanager">[3]</a><cite>
M. W. McLean. “RefManageR: Import and Manage BibTeX and BibLaTeX
References in R”. In: <em>The Journal of Open Source Software</em>
(2017). DOI:
<a href="https://doi.org/10.21105/joss.00338" class="external-link">10.21105/joss.00338</a>.</cite>
</p>
<p>
<a id="bib-ole2024biocstyle"></a><a href="#cite-ole2024biocstyle">[4]</a><cite>
A. Oleś. <em>BiocStyle: Standard styles for vignettes and other
Bioconductor documents</em>. R package version 2.33.1. 2024. DOI:
<a href="https://doi.org/10.18129/B9.bioc.BiocStyle" class="external-link">10.18129/B9.bioc.BiocStyle</a>.
URL:
<a href="https://bioconductor.org/packages/BiocStyle" class="external-link">https://bioconductor.org/packages/BiocStyle</a>.</cite>
</p>
<p>
<a id="bib-2024language"></a><a href="#cite-2024language">[5]</a><cite>
R Core Team. <em>R: A Language and Environment for Statistical
Computing</em>. R Foundation for Statistical Computing. Vienna, Austria,
2024. URL:
<a href="https://www.R-project.org/" class="external-link">https://www.R-project.org/</a>.</cite>
</p>
<p>
<a id="bib-wickham2011testthat"></a><a href="#cite-wickham2011testthat">[6]</a><cite>
H. Wickham. “testthat: Get Started with Testing”. In: <em>The R
Journal</em> 3 (2011), pp. 5–10. URL:
<a href="https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf" class="external-link">https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf</a>.</cite>
</p>
<p>
<a id="bib-wickham2021sessioninfo"></a><a href="#cite-wickham2021sessioninfo">[7]</a><cite>
H. Wickham, W. Chang, R. Flight, et al. <em>sessioninfo: R Session
Information</em>. R package version 1.2.2, <a href="https://r-lib.github.io/sessioninfo/" class="external-link uri">https://r-lib.github.io/sessioninfo/</a>. 2021. URL:
<a href="https://github.com/r-lib/sessioninfo#readme" class="external-link">https://github.com/r-lib/sessioninfo#readme</a>.</cite>
</p>
<p>
<a id="bib-xie2024knitr"></a><a href="#cite-xie2024knitr">[8]</a><cite>
Y. Xie. <em>knitr: A General-Purpose Package for Dynamic Report
Generation in R</em>. R package version 1.48. 2024. URL:
<a href="https://yihui.org/knitr/" class="external-link">https://yihui.org/knitr/</a>.</cite>
</p>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-huuki-myers2024" class="csl-entry">
Huuki-Myers, Louise A., Abby Spangler, Nicholas J. Eagles, Kelsey D.
Montgomery, Sang Ho Kwon, Boyi Guo, Melissa Grant-Peters, et al. 2024.
<span>“A Data-Driven Single-Cell and Spatial Transcriptomic Map of the
Human Prefrontal Cortex.”</span> <em>Science</em> 384 (6698): eadh1938.
<a href="https://doi.org/10.1126/science.adh1938" class="external-link">https://doi.org/10.1126/science.adh1938</a>.
</div>
<div id="ref-lun" class="csl-entry">
Lun, Aaron T. L., Davis J. McCarthy, and John C. Marioni. n.d. <span>“A
Step-by-Step Workflow for Low-Level Analysis of Single-Cell RNA-Seq Data
with Bioconductor.”</span> <a href="https://doi.org/10.12688/f1000research.9501.2" class="external-link">https://doi.org/10.12688/f1000research.9501.2</a>.
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Louise Huuki-Myers.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer>
</div>






  </body>
</html>
